{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Exon-capture to Phylogeny\n",
    "###### Calder Atta - FISH 546 Project\n",
    "---\n",
    "\n",
    "## Workflow\n",
    "1. Examine source data files\n",
    "2. Check quality of reads (FastQC)\n",
    "3. Merge files from multiple lanes\n",
    "4. Remove adapter sequences and low quality score reads (TrimGalore)\n",
    "5. Remove the duplicates from PCR, parse the reads to each locus (custom perl script: preads)\n",
    "6. Assemble the filtered reads into contigs (Trinity)\n",
    "7. Merge the loci containing more than one contigs (Geneious)\n",
    "8. Retrieve orthology by pairwise alignment to corresponding baits sequence (custom perl script: Smith-Waterman algorithm)\n",
    "9. Identify orthology by comparing the retrieved sequence to the genome of O. nilotics (bait source) (Blast)\n",
    "10. Multiple sequences alignment (Clustal Omega v1.1.1)\n",
    "11. Downstream analysis\n",
    "---\n",
    "\n",
    "#### Python Shortcuts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "project = \"/Users/calderatta/Desktop/FISH546_Bioinformatics/project/\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "biotools = \"/Applications/bio-tools/\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "---\n",
    "## 1. Examine source data files\n",
    "\n",
    "#### Location, Files, and Naming\n",
    "The `data/` directory contains reads for 7 species (ID 4 through 10). For each species there are 4 files, representing forward (R1) and reverse (R2) reads for two Illumina lanes (L006 and L008) = 28 files total. Within each forward/reverse pair of .fastq files, the order of sequences is consistent."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[31mTORN_Pool_10_S10_L006_R1_001.fastq\u001b[m\u001b[m* \u001b[31mTORN_Pool_6_S6_L008_R1_001.fastq\u001b[m\u001b[m*\r\n",
      "\u001b[31mTORN_Pool_10_S10_L006_R2_001.fastq\u001b[m\u001b[m* \u001b[31mTORN_Pool_6_S6_L008_R2_001.fastq\u001b[m\u001b[m*\r\n",
      "\u001b[31mTORN_Pool_10_S10_L008_R1_001.fastq\u001b[m\u001b[m* \u001b[31mTORN_Pool_7_S7_L006_R1_001.fastq\u001b[m\u001b[m*\r\n",
      "\u001b[31mTORN_Pool_10_S10_L008_R2_001.fastq\u001b[m\u001b[m* \u001b[31mTORN_Pool_7_S7_L006_R2_001.fastq\u001b[m\u001b[m*\r\n",
      "\u001b[31mTORN_Pool_4_S4_L006_R1_001.fastq\u001b[m\u001b[m*   \u001b[31mTORN_Pool_7_S7_L008_R1_001.fastq\u001b[m\u001b[m*\r\n",
      "\u001b[31mTORN_Pool_4_S4_L006_R2_001.fastq\u001b[m\u001b[m*   \u001b[31mTORN_Pool_7_S7_L008_R2_001.fastq\u001b[m\u001b[m*\r\n",
      "\u001b[31mTORN_Pool_4_S4_L008_R1_001.fastq\u001b[m\u001b[m*   \u001b[31mTORN_Pool_8_S8_L006_R1_001.fastq\u001b[m\u001b[m*\r\n",
      "\u001b[31mTORN_Pool_4_S4_L008_R2_001.fastq\u001b[m\u001b[m*   \u001b[31mTORN_Pool_8_S8_L006_R2_001.fastq\u001b[m\u001b[m*\r\n",
      "\u001b[31mTORN_Pool_5_S5_L006_R1_001.fastq\u001b[m\u001b[m*   \u001b[31mTORN_Pool_8_S8_L008_R1_001.fastq\u001b[m\u001b[m*\r\n",
      "\u001b[31mTORN_Pool_5_S5_L006_R2_001.fastq\u001b[m\u001b[m*   \u001b[31mTORN_Pool_8_S8_L008_R2_001.fastq\u001b[m\u001b[m*\r\n",
      "\u001b[31mTORN_Pool_5_S5_L008_R1_001.fastq\u001b[m\u001b[m*   \u001b[31mTORN_Pool_9_S9_L006_R1_001.fastq\u001b[m\u001b[m*\r\n",
      "\u001b[31mTORN_Pool_5_S5_L008_R2_001.fastq\u001b[m\u001b[m*   \u001b[31mTORN_Pool_9_S9_L006_R2_001.fastq\u001b[m\u001b[m*\r\n",
      "\u001b[31mTORN_Pool_6_S6_L006_R1_001.fastq\u001b[m\u001b[m*   \u001b[31mTORN_Pool_9_S9_L008_R1_001.fastq\u001b[m\u001b[m*\r\n",
      "\u001b[31mTORN_Pool_6_S6_L006_R2_001.fastq\u001b[m\u001b[m*   \u001b[31mTORN_Pool_9_S9_L008_R2_001.fastq\u001b[m\u001b[m*\r\n"
     ]
    }
   ],
   "source": [
    "ls {project}data"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Contents"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "@K00179:70:HHV7JBBXX:6:1101:24454:1209 1:N:0:AACGAAGT\r\n",
      "TNTCTCTCTCTCTTGCTCTCTCTCTCTCTGTTTGAGCTCTCTCTCCCTCTCTCTCTCTCTCTGTCTCTCTCTGTTTGAGCTAACTCTCTCTCTCTGTTAGAGCTCTCTCTCGNTCGCTCTCTCTCGCTNTCGCTGGCTCGCACGCTCTCT\r\n",
      "+\r\n",
      "A#AAAFFFJJJFJFAJFJ-FFFAJFJ-FJ-AA<--7<-7FA7F<A--77A-7A-<777A-<F--A-<FF7F<-----777777F777A-77AF<---7A-----)-7-7-7)#--7--<)F<--7)-)#7A-77)--)7)))7)7)))--\r\n",
      "@K00179:70:HHV7JBBXX:6:1101:26829:1209 1:N:0:AACGAAGT\r\n",
      "CNCTTTCCTTCAGGAGAGACTCTGTCAGGAGGTGCAGGAGGAACAAAAGGAGCAAGAGGAGGAGGATCTGAAGGAGGGATGAGGTGTTGCAGGACGATGAACAGGAGGGGGAGCATGAGGAGGAGCAGGAGTAGGTGGAGCATAAGGAGG\r\n",
      "+\r\n",
      "A#AAFFJFFJJJJJJAFJJJAJFJ<JAAJJJJAJJJ7FFF<AJJFAJFFJ<AFJJF<AJJJFFJA77F7A--F-AAFF-7FJJJAF7FJ<AAFFJJ77AF7<<A<AJF))-))-)-77A<-<7AJF)<7)7-7-7<F-))7)---7--7)\r\n",
      "@K00179:70:HHV7JBBXX:6:1101:4472:1226 1:N:0:AACGAAGT\r\n",
      "GNAACAACATGGAGGTCAGAGGAGGAACAACATGGAGGTCAGAGGAGGAACAACATGGAGGTCAGAGGAGGAACAACATGGAGGTCAGAGAAGGCGCATCACGTATCTCAGANGAAAAGAAAGGAGGTNTGCAAAGACGAACGAGGGGGC\r\n"
     ]
    }
   ],
   "source": [
    "!head {project}data/TORN_Pool_10_S10_L006_R1_001.fastq"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Sequence Counts\n",
    "Now let's check that the number of sequences in R1 and R2 files match. First test on one file, then apply to the rest."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2419885\r\n"
     ]
    }
   ],
   "source": [
    "!grep -c \"@\" {project}data/TORN_Pool_10_S10_L008_R2_001.fastq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/calderatta/Desktop/FISH 546 - Bioinformatics/project/raw/TORN_Pool_10_S10_L006_R1_001.fastq:2419885\n",
      "/Users/calderatta/Desktop/FISH 546 - Bioinformatics/project/raw/TORN_Pool_10_S10_L006_R2_001.fastq:2419885\n",
      "/Users/calderatta/Desktop/FISH 546 - Bioinformatics/project/raw/TORN_Pool_10_S10_L008_R1_001.fastq:1497568\n",
      "/Users/calderatta/Desktop/FISH 546 - Bioinformatics/project/raw/TORN_Pool_10_S10_L008_R2_001.fastq:1497568\n",
      "/Users/calderatta/Desktop/FISH 546 - Bioinformatics/project/raw/TORN_Pool_4_S4_L006_R1_001.fastq:3061218\n",
      "/Users/calderatta/Desktop/FISH 546 - Bioinformatics/project/raw/TORN_Pool_4_S4_L006_R2_001.fastq:3061218\n",
      "/Users/calderatta/Desktop/FISH 546 - Bioinformatics/project/raw/TORN_Pool_4_S4_L008_R1_001.fastq:2506641\n",
      "/Users/calderatta/Desktop/FISH 546 - Bioinformatics/project/raw/TORN_Pool_4_S4_L008_R2_001.fastq:2506641\n",
      "/Users/calderatta/Desktop/FISH 546 - Bioinformatics/project/raw/TORN_Pool_5_S5_L006_R1_001.fastq:1248025\n",
      "/Users/calderatta/Desktop/FISH 546 - Bioinformatics/project/raw/TORN_Pool_5_S5_L006_R2_001.fastq:1248025\n",
      "/Users/calderatta/Desktop/FISH 546 - Bioinformatics/project/raw/TORN_Pool_5_S5_L008_R1_001.fastq:870104\n",
      "/Users/calderatta/Desktop/FISH 546 - Bioinformatics/project/raw/TORN_Pool_5_S5_L008_R2_001.fastq:870104\n",
      "/Users/calderatta/Desktop/FISH 546 - Bioinformatics/project/raw/TORN_Pool_6_S6_L006_R1_001.fastq:3406559\n",
      "/Users/calderatta/Desktop/FISH 546 - Bioinformatics/project/raw/TORN_Pool_6_S6_L006_R2_001.fastq:3406559\n",
      "/Users/calderatta/Desktop/FISH 546 - Bioinformatics/project/raw/TORN_Pool_6_S6_L008_R1_001.fastq:3107867\n",
      "/Users/calderatta/Desktop/FISH 546 - Bioinformatics/project/raw/TORN_Pool_6_S6_L008_R2_001.fastq:3107867\n",
      "/Users/calderatta/Desktop/FISH 546 - Bioinformatics/project/raw/TORN_Pool_7_S7_L006_R1_001.fastq:2565256\n",
      "/Users/calderatta/Desktop/FISH 546 - Bioinformatics/project/raw/TORN_Pool_7_S7_L006_R2_001.fastq:2565256\n",
      "/Users/calderatta/Desktop/FISH 546 - Bioinformatics/project/raw/TORN_Pool_7_S7_L008_R1_001.fastq:3177384\n",
      "/Users/calderatta/Desktop/FISH 546 - Bioinformatics/project/raw/TORN_Pool_7_S7_L008_R2_001.fastq:3177384\n",
      "/Users/calderatta/Desktop/FISH 546 - Bioinformatics/project/raw/TORN_Pool_8_S8_L006_R1_001.fastq:2083571\n",
      "/Users/calderatta/Desktop/FISH 546 - Bioinformatics/project/raw/TORN_Pool_8_S8_L006_R2_001.fastq:2083571\n",
      "/Users/calderatta/Desktop/FISH 546 - Bioinformatics/project/raw/TORN_Pool_8_S8_L008_R1_001.fastq:2274644\n",
      "/Users/calderatta/Desktop/FISH 546 - Bioinformatics/project/raw/TORN_Pool_8_S8_L008_R2_001.fastq:2274644\n",
      "/Users/calderatta/Desktop/FISH 546 - Bioinformatics/project/raw/TORN_Pool_9_S9_L006_R1_001.fastq:2924224\n",
      "/Users/calderatta/Desktop/FISH 546 - Bioinformatics/project/raw/TORN_Pool_9_S9_L006_R2_001.fastq:2924224\n",
      "/Users/calderatta/Desktop/FISH 546 - Bioinformatics/project/raw/TORN_Pool_9_S9_L008_R1_001.fastq:2114997\n",
      "/Users/calderatta/Desktop/FISH 546 - Bioinformatics/project/raw/TORN_Pool_9_S9_L008_R2_001.fastq:2114997\n"
     ]
    }
   ],
   "source": [
    "!grep -c '@' {project}data/*.fastq"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "They match!\n",
    "## 2. Check quality of reads (FastQC)\n",
    "Objective: Examine quality of reads for each sequence file. \n",
    "\n",
    "Input: source files (.fastq)\n",
    "\n",
    "    {project}/data/*.fastq              (eg. TORN_Pool_4_S4_L006_R1_001.fastqc)\n",
    "Output: sequence quality plots (.html/.zip)\n",
    "\n",
    "    {project}/analysis/fastqc/*.fastq   (eg. TORN_Pool_4_S4_L006_R1_001_fastqc.html)\n",
    "    {project}/analysis/fastqc/*.fastq   (eg. TORN_Pool_4_S4_L006_R1_001_fastqc.zip)\n",
    "Requires:\n",
    "- fastqc (See `notebooks/installation.ipynb` for installation instructions.)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Started analysis of TORN_Pool_10_S10_L006_R1_001.fastq\n",
      "Approx 5% complete for TORN_Pool_10_S10_L006_R1_001.fastq\n",
      "Approx 10% complete for TORN_Pool_10_S10_L006_R1_001.fastq\n",
      "Approx 15% complete for TORN_Pool_10_S10_L006_R1_001.fastq\n",
      "Approx 20% complete for TORN_Pool_10_S10_L006_R1_001.fastq\n",
      "Approx 25% complete for TORN_Pool_10_S10_L006_R1_001.fastq\n",
      "Approx 30% complete for TORN_Pool_10_S10_L006_R1_001.fastq\n",
      "Approx 35% complete for TORN_Pool_10_S10_L006_R1_001.fastq\n",
      "Approx 40% complete for TORN_Pool_10_S10_L006_R1_001.fastq\n",
      "Approx 45% complete for TORN_Pool_10_S10_L006_R1_001.fastq\n",
      "Approx 50% complete for TORN_Pool_10_S10_L006_R1_001.fastq\n",
      "Approx 55% complete for TORN_Pool_10_S10_L006_R1_001.fastq\n",
      "Approx 60% complete for TORN_Pool_10_S10_L006_R1_001.fastq\n",
      "Approx 65% complete for TORN_Pool_10_S10_L006_R1_001.fastq\n",
      "Approx 70% complete for TORN_Pool_10_S10_L006_R1_001.fastq\n",
      "Approx 75% complete for TORN_Pool_10_S10_L006_R1_001.fastq\n",
      "Approx 80% complete for TORN_Pool_10_S10_L006_R1_001.fastq\n",
      "Approx 85% complete for TORN_Pool_10_S10_L006_R1_001.fastq\n",
      "Approx 90% complete for TORN_Pool_10_S10_L006_R1_001.fastq\n",
      "Approx 95% complete for TORN_Pool_10_S10_L006_R1_001.fastq\n",
      "Analysis complete for TORN_Pool_10_S10_L006_R1_001.fastq\n",
      "Started analysis of TORN_Pool_10_S10_L006_R2_001.fastq\n",
      "Approx 5% complete for TORN_Pool_10_S10_L006_R2_001.fastq\n",
      "Approx 10% complete for TORN_Pool_10_S10_L006_R2_001.fastq\n",
      "Approx 15% complete for TORN_Pool_10_S10_L006_R2_001.fastq\n",
      "Approx 20% complete for TORN_Pool_10_S10_L006_R2_001.fastq\n",
      "Approx 25% complete for TORN_Pool_10_S10_L006_R2_001.fastq\n",
      "Approx 30% complete for TORN_Pool_10_S10_L006_R2_001.fastq\n",
      "Approx 35% complete for TORN_Pool_10_S10_L006_R2_001.fastq\n",
      "Approx 40% complete for TORN_Pool_10_S10_L006_R2_001.fastq\n",
      "Approx 45% complete for TORN_Pool_10_S10_L006_R2_001.fastq\n",
      "Approx 50% complete for TORN_Pool_10_S10_L006_R2_001.fastq\n",
      "Approx 55% complete for TORN_Pool_10_S10_L006_R2_001.fastq\n",
      "Approx 60% complete for TORN_Pool_10_S10_L006_R2_001.fastq\n",
      "Approx 65% complete for TORN_Pool_10_S10_L006_R2_001.fastq\n",
      "Approx 70% complete for TORN_Pool_10_S10_L006_R2_001.fastq\n",
      "Approx 75% complete for TORN_Pool_10_S10_L006_R2_001.fastq\n",
      "Approx 80% complete for TORN_Pool_10_S10_L006_R2_001.fastq\n",
      "Approx 85% complete for TORN_Pool_10_S10_L006_R2_001.fastq\n",
      "Approx 90% complete for TORN_Pool_10_S10_L006_R2_001.fastq\n",
      "Approx 95% complete for TORN_Pool_10_S10_L006_R2_001.fastq\n",
      "Analysis complete for TORN_Pool_10_S10_L006_R2_001.fastq\n",
      "Started analysis of TORN_Pool_10_S10_L008_R1_001.fastq\n",
      "Approx 5% complete for TORN_Pool_10_S10_L008_R1_001.fastq\n",
      "Approx 10% complete for TORN_Pool_10_S10_L008_R1_001.fastq\n",
      "Approx 15% complete for TORN_Pool_10_S10_L008_R1_001.fastq\n",
      "Approx 20% complete for TORN_Pool_10_S10_L008_R1_001.fastq\n",
      "Approx 25% complete for TORN_Pool_10_S10_L008_R1_001.fastq\n",
      "Approx 30% complete for TORN_Pool_10_S10_L008_R1_001.fastq\n",
      "Approx 35% complete for TORN_Pool_10_S10_L008_R1_001.fastq\n",
      "Approx 40% complete for TORN_Pool_10_S10_L008_R1_001.fastq\n",
      "Approx 45% complete for TORN_Pool_10_S10_L008_R1_001.fastq\n",
      "Approx 50% complete for TORN_Pool_10_S10_L008_R1_001.fastq\n",
      "Approx 55% complete for TORN_Pool_10_S10_L008_R1_001.fastq\n",
      "Approx 60% complete for TORN_Pool_10_S10_L008_R1_001.fastq\n",
      "Approx 65% complete for TORN_Pool_10_S10_L008_R1_001.fastq\n",
      "Approx 70% complete for TORN_Pool_10_S10_L008_R1_001.fastq\n",
      "Approx 75% complete for TORN_Pool_10_S10_L008_R1_001.fastq\n",
      "Approx 80% complete for TORN_Pool_10_S10_L008_R1_001.fastq\n",
      "Approx 85% complete for TORN_Pool_10_S10_L008_R1_001.fastq\n",
      "Approx 90% complete for TORN_Pool_10_S10_L008_R1_001.fastq\n",
      "Approx 95% complete for TORN_Pool_10_S10_L008_R1_001.fastq\n",
      "Analysis complete for TORN_Pool_10_S10_L008_R1_001.fastq\n",
      "Started analysis of TORN_Pool_10_S10_L008_R2_001.fastq\n",
      "Approx 5% complete for TORN_Pool_10_S10_L008_R2_001.fastq\n",
      "Approx 10% complete for TORN_Pool_10_S10_L008_R2_001.fastq\n",
      "Approx 15% complete for TORN_Pool_10_S10_L008_R2_001.fastq\n",
      "Approx 20% complete for TORN_Pool_10_S10_L008_R2_001.fastq\n",
      "Approx 25% complete for TORN_Pool_10_S10_L008_R2_001.fastq\n",
      "Approx 30% complete for TORN_Pool_10_S10_L008_R2_001.fastq\n",
      "Approx 35% complete for TORN_Pool_10_S10_L008_R2_001.fastq\n",
      "Approx 40% complete for TORN_Pool_10_S10_L008_R2_001.fastq\n",
      "Approx 45% complete for TORN_Pool_10_S10_L008_R2_001.fastq\n",
      "Approx 50% complete for TORN_Pool_10_S10_L008_R2_001.fastq\n",
      "Approx 55% complete for TORN_Pool_10_S10_L008_R2_001.fastq\n",
      "Approx 60% complete for TORN_Pool_10_S10_L008_R2_001.fastq\n",
      "Approx 65% complete for TORN_Pool_10_S10_L008_R2_001.fastq\n",
      "Approx 70% complete for TORN_Pool_10_S10_L008_R2_001.fastq\n",
      "Approx 75% complete for TORN_Pool_10_S10_L008_R2_001.fastq\n",
      "Approx 80% complete for TORN_Pool_10_S10_L008_R2_001.fastq\n",
      "Approx 85% complete for TORN_Pool_10_S10_L008_R2_001.fastq\n",
      "Approx 90% complete for TORN_Pool_10_S10_L008_R2_001.fastq\n",
      "Approx 95% complete for TORN_Pool_10_S10_L008_R2_001.fastq\n",
      "Analysis complete for TORN_Pool_10_S10_L008_R2_001.fastq\n",
      "Started analysis of TORN_Pool_4_S4_L006_R1_001.fastq\n",
      "Approx 5% complete for TORN_Pool_4_S4_L006_R1_001.fastq\n",
      "Approx 10% complete for TORN_Pool_4_S4_L006_R1_001.fastq\n",
      "Approx 15% complete for TORN_Pool_4_S4_L006_R1_001.fastq\n",
      "Approx 20% complete for TORN_Pool_4_S4_L006_R1_001.fastq\n",
      "Approx 25% complete for TORN_Pool_4_S4_L006_R1_001.fastq\n",
      "Approx 30% complete for TORN_Pool_4_S4_L006_R1_001.fastq\n",
      "Approx 35% complete for TORN_Pool_4_S4_L006_R1_001.fastq\n",
      "Approx 40% complete for TORN_Pool_4_S4_L006_R1_001.fastq\n",
      "Approx 45% complete for TORN_Pool_4_S4_L006_R1_001.fastq\n",
      "Approx 50% complete for TORN_Pool_4_S4_L006_R1_001.fastq\n",
      "Approx 55% complete for TORN_Pool_4_S4_L006_R1_001.fastq\n",
      "Approx 60% complete for TORN_Pool_4_S4_L006_R1_001.fastq\n",
      "Approx 65% complete for TORN_Pool_4_S4_L006_R1_001.fastq\n",
      "Approx 70% complete for TORN_Pool_4_S4_L006_R1_001.fastq\n",
      "Approx 75% complete for TORN_Pool_4_S4_L006_R1_001.fastq\n",
      "Approx 80% complete for TORN_Pool_4_S4_L006_R1_001.fastq\n",
      "Approx 85% complete for TORN_Pool_4_S4_L006_R1_001.fastq\n",
      "Approx 90% complete for TORN_Pool_4_S4_L006_R1_001.fastq\n",
      "Approx 95% complete for TORN_Pool_4_S4_L006_R1_001.fastq\n",
      "Analysis complete for TORN_Pool_4_S4_L006_R1_001.fastq\n",
      "Started analysis of TORN_Pool_4_S4_L006_R2_001.fastq\n",
      "Approx 5% complete for TORN_Pool_4_S4_L006_R2_001.fastq\n",
      "Approx 10% complete for TORN_Pool_4_S4_L006_R2_001.fastq\n",
      "Approx 15% complete for TORN_Pool_4_S4_L006_R2_001.fastq\n",
      "Approx 20% complete for TORN_Pool_4_S4_L006_R2_001.fastq\n",
      "Approx 25% complete for TORN_Pool_4_S4_L006_R2_001.fastq\n",
      "Approx 30% complete for TORN_Pool_4_S4_L006_R2_001.fastq\n",
      "Approx 35% complete for TORN_Pool_4_S4_L006_R2_001.fastq\n",
      "Approx 40% complete for TORN_Pool_4_S4_L006_R2_001.fastq\n",
      "Approx 45% complete for TORN_Pool_4_S4_L006_R2_001.fastq\n",
      "Approx 50% complete for TORN_Pool_4_S4_L006_R2_001.fastq\n",
      "Approx 55% complete for TORN_Pool_4_S4_L006_R2_001.fastq\n",
      "Approx 60% complete for TORN_Pool_4_S4_L006_R2_001.fastq\n",
      "Approx 65% complete for TORN_Pool_4_S4_L006_R2_001.fastq\n",
      "Approx 70% complete for TORN_Pool_4_S4_L006_R2_001.fastq\n",
      "Approx 75% complete for TORN_Pool_4_S4_L006_R2_001.fastq\n",
      "Approx 80% complete for TORN_Pool_4_S4_L006_R2_001.fastq\n",
      "Approx 85% complete for TORN_Pool_4_S4_L006_R2_001.fastq\n",
      "Approx 90% complete for TORN_Pool_4_S4_L006_R2_001.fastq\n",
      "Approx 95% complete for TORN_Pool_4_S4_L006_R2_001.fastq\n",
      "Analysis complete for TORN_Pool_4_S4_L006_R2_001.fastq\n",
      "Started analysis of TORN_Pool_4_S4_L008_R1_001.fastq\n",
      "Approx 5% complete for TORN_Pool_4_S4_L008_R1_001.fastq\n",
      "Approx 10% complete for TORN_Pool_4_S4_L008_R1_001.fastq\n",
      "Approx 15% complete for TORN_Pool_4_S4_L008_R1_001.fastq\n",
      "Approx 20% complete for TORN_Pool_4_S4_L008_R1_001.fastq\n",
      "Approx 25% complete for TORN_Pool_4_S4_L008_R1_001.fastq\n",
      "Approx 30% complete for TORN_Pool_4_S4_L008_R1_001.fastq\n",
      "Approx 35% complete for TORN_Pool_4_S4_L008_R1_001.fastq\n",
      "Approx 40% complete for TORN_Pool_4_S4_L008_R1_001.fastq\n",
      "Approx 45% complete for TORN_Pool_4_S4_L008_R1_001.fastq\n",
      "Approx 50% complete for TORN_Pool_4_S4_L008_R1_001.fastq\n",
      "Approx 55% complete for TORN_Pool_4_S4_L008_R1_001.fastq\n",
      "Approx 60% complete for TORN_Pool_4_S4_L008_R1_001.fastq\n",
      "Approx 65% complete for TORN_Pool_4_S4_L008_R1_001.fastq\n",
      "Approx 70% complete for TORN_Pool_4_S4_L008_R1_001.fastq\n",
      "Approx 75% complete for TORN_Pool_4_S4_L008_R1_001.fastq\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Approx 80% complete for TORN_Pool_4_S4_L008_R1_001.fastq\n",
      "Approx 85% complete for TORN_Pool_4_S4_L008_R1_001.fastq\n",
      "Approx 90% complete for TORN_Pool_4_S4_L008_R1_001.fastq\n",
      "Approx 95% complete for TORN_Pool_4_S4_L008_R1_001.fastq\n",
      "Analysis complete for TORN_Pool_4_S4_L008_R1_001.fastq\n",
      "Started analysis of TORN_Pool_4_S4_L008_R2_001.fastq\n",
      "Approx 5% complete for TORN_Pool_4_S4_L008_R2_001.fastq\n",
      "Approx 10% complete for TORN_Pool_4_S4_L008_R2_001.fastq\n",
      "Approx 15% complete for TORN_Pool_4_S4_L008_R2_001.fastq\n",
      "Approx 20% complete for TORN_Pool_4_S4_L008_R2_001.fastq\n",
      "Approx 25% complete for TORN_Pool_4_S4_L008_R2_001.fastq\n",
      "Approx 30% complete for TORN_Pool_4_S4_L008_R2_001.fastq\n",
      "Approx 35% complete for TORN_Pool_4_S4_L008_R2_001.fastq\n",
      "Approx 40% complete for TORN_Pool_4_S4_L008_R2_001.fastq\n",
      "Approx 45% complete for TORN_Pool_4_S4_L008_R2_001.fastq\n",
      "Approx 50% complete for TORN_Pool_4_S4_L008_R2_001.fastq\n",
      "Approx 55% complete for TORN_Pool_4_S4_L008_R2_001.fastq\n",
      "Approx 60% complete for TORN_Pool_4_S4_L008_R2_001.fastq\n",
      "Approx 65% complete for TORN_Pool_4_S4_L008_R2_001.fastq\n",
      "Approx 70% complete for TORN_Pool_4_S4_L008_R2_001.fastq\n",
      "Approx 75% complete for TORN_Pool_4_S4_L008_R2_001.fastq\n",
      "Approx 80% complete for TORN_Pool_4_S4_L008_R2_001.fastq\n",
      "Approx 85% complete for TORN_Pool_4_S4_L008_R2_001.fastq\n",
      "Approx 90% complete for TORN_Pool_4_S4_L008_R2_001.fastq\n",
      "Approx 95% complete for TORN_Pool_4_S4_L008_R2_001.fastq\n",
      "Analysis complete for TORN_Pool_4_S4_L008_R2_001.fastq\n",
      "Started analysis of TORN_Pool_5_S5_L006_R1_001.fastq\n",
      "Approx 5% complete for TORN_Pool_5_S5_L006_R1_001.fastq\n",
      "Approx 10% complete for TORN_Pool_5_S5_L006_R1_001.fastq\n",
      "Approx 15% complete for TORN_Pool_5_S5_L006_R1_001.fastq\n",
      "Approx 20% complete for TORN_Pool_5_S5_L006_R1_001.fastq\n",
      "Approx 25% complete for TORN_Pool_5_S5_L006_R1_001.fastq\n",
      "Approx 30% complete for TORN_Pool_5_S5_L006_R1_001.fastq\n",
      "Approx 35% complete for TORN_Pool_5_S5_L006_R1_001.fastq\n",
      "Approx 40% complete for TORN_Pool_5_S5_L006_R1_001.fastq\n",
      "Approx 45% complete for TORN_Pool_5_S5_L006_R1_001.fastq\n",
      "Approx 50% complete for TORN_Pool_5_S5_L006_R1_001.fastq\n",
      "Approx 55% complete for TORN_Pool_5_S5_L006_R1_001.fastq\n",
      "Approx 60% complete for TORN_Pool_5_S5_L006_R1_001.fastq\n",
      "Approx 65% complete for TORN_Pool_5_S5_L006_R1_001.fastq\n",
      "Approx 70% complete for TORN_Pool_5_S5_L006_R1_001.fastq\n",
      "Approx 75% complete for TORN_Pool_5_S5_L006_R1_001.fastq\n",
      "Approx 80% complete for TORN_Pool_5_S5_L006_R1_001.fastq\n",
      "Approx 85% complete for TORN_Pool_5_S5_L006_R1_001.fastq\n",
      "Approx 90% complete for TORN_Pool_5_S5_L006_R1_001.fastq\n",
      "Approx 95% complete for TORN_Pool_5_S5_L006_R1_001.fastq\n",
      "Analysis complete for TORN_Pool_5_S5_L006_R1_001.fastq\n",
      "Started analysis of TORN_Pool_5_S5_L006_R2_001.fastq\n",
      "Approx 5% complete for TORN_Pool_5_S5_L006_R2_001.fastq\n",
      "Approx 10% complete for TORN_Pool_5_S5_L006_R2_001.fastq\n",
      "Approx 15% complete for TORN_Pool_5_S5_L006_R2_001.fastq\n",
      "Approx 20% complete for TORN_Pool_5_S5_L006_R2_001.fastq\n",
      "Approx 25% complete for TORN_Pool_5_S5_L006_R2_001.fastq\n",
      "Approx 30% complete for TORN_Pool_5_S5_L006_R2_001.fastq\n",
      "Approx 35% complete for TORN_Pool_5_S5_L006_R2_001.fastq\n",
      "Approx 40% complete for TORN_Pool_5_S5_L006_R2_001.fastq\n",
      "Approx 45% complete for TORN_Pool_5_S5_L006_R2_001.fastq\n",
      "Approx 50% complete for TORN_Pool_5_S5_L006_R2_001.fastq\n",
      "Approx 55% complete for TORN_Pool_5_S5_L006_R2_001.fastq\n",
      "Approx 60% complete for TORN_Pool_5_S5_L006_R2_001.fastq\n",
      "Approx 65% complete for TORN_Pool_5_S5_L006_R2_001.fastq\n",
      "Approx 70% complete for TORN_Pool_5_S5_L006_R2_001.fastq\n",
      "Approx 75% complete for TORN_Pool_5_S5_L006_R2_001.fastq\n",
      "Approx 80% complete for TORN_Pool_5_S5_L006_R2_001.fastq\n",
      "Approx 85% complete for TORN_Pool_5_S5_L006_R2_001.fastq\n",
      "Approx 90% complete for TORN_Pool_5_S5_L006_R2_001.fastq\n",
      "Approx 95% complete for TORN_Pool_5_S5_L006_R2_001.fastq\n",
      "Analysis complete for TORN_Pool_5_S5_L006_R2_001.fastq\n",
      "Started analysis of TORN_Pool_5_S5_L008_R1_001.fastq\n",
      "Approx 5% complete for TORN_Pool_5_S5_L008_R1_001.fastq\n",
      "Approx 10% complete for TORN_Pool_5_S5_L008_R1_001.fastq\n",
      "Approx 15% complete for TORN_Pool_5_S5_L008_R1_001.fastq\n",
      "Approx 20% complete for TORN_Pool_5_S5_L008_R1_001.fastq\n",
      "Approx 25% complete for TORN_Pool_5_S5_L008_R1_001.fastq\n",
      "Approx 30% complete for TORN_Pool_5_S5_L008_R1_001.fastq\n",
      "Approx 35% complete for TORN_Pool_5_S5_L008_R1_001.fastq\n",
      "Approx 40% complete for TORN_Pool_5_S5_L008_R1_001.fastq\n",
      "Approx 45% complete for TORN_Pool_5_S5_L008_R1_001.fastq\n",
      "Approx 50% complete for TORN_Pool_5_S5_L008_R1_001.fastq\n",
      "Approx 55% complete for TORN_Pool_5_S5_L008_R1_001.fastq\n",
      "Approx 60% complete for TORN_Pool_5_S5_L008_R1_001.fastq\n",
      "Approx 65% complete for TORN_Pool_5_S5_L008_R1_001.fastq\n",
      "Approx 70% complete for TORN_Pool_5_S5_L008_R1_001.fastq\n",
      "Approx 75% complete for TORN_Pool_5_S5_L008_R1_001.fastq\n",
      "Approx 80% complete for TORN_Pool_5_S5_L008_R1_001.fastq\n",
      "Approx 85% complete for TORN_Pool_5_S5_L008_R1_001.fastq\n",
      "Approx 90% complete for TORN_Pool_5_S5_L008_R1_001.fastq\n",
      "Approx 95% complete for TORN_Pool_5_S5_L008_R1_001.fastq\n",
      "Analysis complete for TORN_Pool_5_S5_L008_R1_001.fastq\n",
      "Started analysis of TORN_Pool_5_S5_L008_R2_001.fastq\n",
      "Approx 5% complete for TORN_Pool_5_S5_L008_R2_001.fastq\n",
      "Approx 10% complete for TORN_Pool_5_S5_L008_R2_001.fastq\n",
      "Approx 15% complete for TORN_Pool_5_S5_L008_R2_001.fastq\n",
      "Approx 20% complete for TORN_Pool_5_S5_L008_R2_001.fastq\n",
      "Approx 25% complete for TORN_Pool_5_S5_L008_R2_001.fastq\n",
      "Approx 30% complete for TORN_Pool_5_S5_L008_R2_001.fastq\n",
      "Approx 35% complete for TORN_Pool_5_S5_L008_R2_001.fastq\n",
      "Approx 40% complete for TORN_Pool_5_S5_L008_R2_001.fastq\n",
      "Approx 45% complete for TORN_Pool_5_S5_L008_R2_001.fastq\n",
      "Approx 50% complete for TORN_Pool_5_S5_L008_R2_001.fastq\n",
      "Approx 55% complete for TORN_Pool_5_S5_L008_R2_001.fastq\n",
      "Approx 60% complete for TORN_Pool_5_S5_L008_R2_001.fastq\n",
      "Approx 65% complete for TORN_Pool_5_S5_L008_R2_001.fastq\n",
      "Approx 70% complete for TORN_Pool_5_S5_L008_R2_001.fastq\n",
      "Approx 75% complete for TORN_Pool_5_S5_L008_R2_001.fastq\n",
      "Approx 80% complete for TORN_Pool_5_S5_L008_R2_001.fastq\n",
      "Approx 85% complete for TORN_Pool_5_S5_L008_R2_001.fastq\n",
      "Approx 90% complete for TORN_Pool_5_S5_L008_R2_001.fastq\n",
      "Approx 95% complete for TORN_Pool_5_S5_L008_R2_001.fastq\n",
      "Analysis complete for TORN_Pool_5_S5_L008_R2_001.fastq\n",
      "Started analysis of TORN_Pool_6_S6_L006_R1_001.fastq\n",
      "Approx 5% complete for TORN_Pool_6_S6_L006_R1_001.fastq\n",
      "Approx 10% complete for TORN_Pool_6_S6_L006_R1_001.fastq\n",
      "Approx 15% complete for TORN_Pool_6_S6_L006_R1_001.fastq\n",
      "Approx 20% complete for TORN_Pool_6_S6_L006_R1_001.fastq\n",
      "Approx 25% complete for TORN_Pool_6_S6_L006_R1_001.fastq\n",
      "Approx 30% complete for TORN_Pool_6_S6_L006_R1_001.fastq\n",
      "Approx 35% complete for TORN_Pool_6_S6_L006_R1_001.fastq\n",
      "Approx 40% complete for TORN_Pool_6_S6_L006_R1_001.fastq\n",
      "Approx 45% complete for TORN_Pool_6_S6_L006_R1_001.fastq\n",
      "Approx 50% complete for TORN_Pool_6_S6_L006_R1_001.fastq\n",
      "Approx 55% complete for TORN_Pool_6_S6_L006_R1_001.fastq\n",
      "Approx 60% complete for TORN_Pool_6_S6_L006_R1_001.fastq\n",
      "Approx 65% complete for TORN_Pool_6_S6_L006_R1_001.fastq\n",
      "Approx 70% complete for TORN_Pool_6_S6_L006_R1_001.fastq\n",
      "Approx 75% complete for TORN_Pool_6_S6_L006_R1_001.fastq\n",
      "Approx 80% complete for TORN_Pool_6_S6_L006_R1_001.fastq\n",
      "Approx 85% complete for TORN_Pool_6_S6_L006_R1_001.fastq\n",
      "Approx 90% complete for TORN_Pool_6_S6_L006_R1_001.fastq\n",
      "Approx 95% complete for TORN_Pool_6_S6_L006_R1_001.fastq\n",
      "Analysis complete for TORN_Pool_6_S6_L006_R1_001.fastq\n",
      "Started analysis of TORN_Pool_6_S6_L006_R2_001.fastq\n",
      "Approx 5% complete for TORN_Pool_6_S6_L006_R2_001.fastq\n",
      "Approx 10% complete for TORN_Pool_6_S6_L006_R2_001.fastq\n",
      "Approx 15% complete for TORN_Pool_6_S6_L006_R2_001.fastq\n",
      "Approx 20% complete for TORN_Pool_6_S6_L006_R2_001.fastq\n",
      "Approx 25% complete for TORN_Pool_6_S6_L006_R2_001.fastq\n",
      "Approx 30% complete for TORN_Pool_6_S6_L006_R2_001.fastq\n",
      "Approx 35% complete for TORN_Pool_6_S6_L006_R2_001.fastq\n",
      "Approx 40% complete for TORN_Pool_6_S6_L006_R2_001.fastq\n",
      "Approx 45% complete for TORN_Pool_6_S6_L006_R2_001.fastq\n",
      "Approx 50% complete for TORN_Pool_6_S6_L006_R2_001.fastq\n",
      "Approx 55% complete for TORN_Pool_6_S6_L006_R2_001.fastq\n",
      "Approx 60% complete for TORN_Pool_6_S6_L006_R2_001.fastq\n",
      "Approx 65% complete for TORN_Pool_6_S6_L006_R2_001.fastq\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Approx 70% complete for TORN_Pool_6_S6_L006_R2_001.fastq\n",
      "Approx 75% complete for TORN_Pool_6_S6_L006_R2_001.fastq\n",
      "Approx 80% complete for TORN_Pool_6_S6_L006_R2_001.fastq\n",
      "Approx 85% complete for TORN_Pool_6_S6_L006_R2_001.fastq\n",
      "Approx 90% complete for TORN_Pool_6_S6_L006_R2_001.fastq\n",
      "Approx 95% complete for TORN_Pool_6_S6_L006_R2_001.fastq\n",
      "Analysis complete for TORN_Pool_6_S6_L006_R2_001.fastq\n",
      "Started analysis of TORN_Pool_6_S6_L008_R1_001.fastq\n",
      "Approx 5% complete for TORN_Pool_6_S6_L008_R1_001.fastq\n",
      "Approx 10% complete for TORN_Pool_6_S6_L008_R1_001.fastq\n",
      "Approx 15% complete for TORN_Pool_6_S6_L008_R1_001.fastq\n",
      "Approx 20% complete for TORN_Pool_6_S6_L008_R1_001.fastq\n",
      "Approx 25% complete for TORN_Pool_6_S6_L008_R1_001.fastq\n",
      "Approx 30% complete for TORN_Pool_6_S6_L008_R1_001.fastq\n",
      "Approx 35% complete for TORN_Pool_6_S6_L008_R1_001.fastq\n",
      "Approx 40% complete for TORN_Pool_6_S6_L008_R1_001.fastq\n",
      "Approx 45% complete for TORN_Pool_6_S6_L008_R1_001.fastq\n",
      "Approx 50% complete for TORN_Pool_6_S6_L008_R1_001.fastq\n",
      "Approx 55% complete for TORN_Pool_6_S6_L008_R1_001.fastq\n",
      "Approx 60% complete for TORN_Pool_6_S6_L008_R1_001.fastq\n",
      "Approx 65% complete for TORN_Pool_6_S6_L008_R1_001.fastq\n",
      "Approx 70% complete for TORN_Pool_6_S6_L008_R1_001.fastq\n",
      "Approx 75% complete for TORN_Pool_6_S6_L008_R1_001.fastq\n",
      "Approx 80% complete for TORN_Pool_6_S6_L008_R1_001.fastq\n",
      "Approx 85% complete for TORN_Pool_6_S6_L008_R1_001.fastq\n",
      "Approx 90% complete for TORN_Pool_6_S6_L008_R1_001.fastq\n",
      "Approx 95% complete for TORN_Pool_6_S6_L008_R1_001.fastq\n",
      "Analysis complete for TORN_Pool_6_S6_L008_R1_001.fastq\n",
      "Started analysis of TORN_Pool_6_S6_L008_R2_001.fastq\n",
      "Approx 5% complete for TORN_Pool_6_S6_L008_R2_001.fastq\n",
      "Approx 10% complete for TORN_Pool_6_S6_L008_R2_001.fastq\n",
      "Approx 15% complete for TORN_Pool_6_S6_L008_R2_001.fastq\n",
      "Approx 20% complete for TORN_Pool_6_S6_L008_R2_001.fastq\n",
      "Approx 25% complete for TORN_Pool_6_S6_L008_R2_001.fastq\n",
      "Approx 30% complete for TORN_Pool_6_S6_L008_R2_001.fastq\n",
      "Approx 35% complete for TORN_Pool_6_S6_L008_R2_001.fastq\n",
      "Approx 40% complete for TORN_Pool_6_S6_L008_R2_001.fastq\n",
      "Approx 45% complete for TORN_Pool_6_S6_L008_R2_001.fastq\n",
      "Approx 50% complete for TORN_Pool_6_S6_L008_R2_001.fastq\n",
      "Approx 55% complete for TORN_Pool_6_S6_L008_R2_001.fastq\n",
      "Approx 60% complete for TORN_Pool_6_S6_L008_R2_001.fastq\n",
      "Approx 65% complete for TORN_Pool_6_S6_L008_R2_001.fastq\n",
      "Approx 70% complete for TORN_Pool_6_S6_L008_R2_001.fastq\n",
      "Approx 75% complete for TORN_Pool_6_S6_L008_R2_001.fastq\n",
      "Approx 80% complete for TORN_Pool_6_S6_L008_R2_001.fastq\n",
      "Approx 85% complete for TORN_Pool_6_S6_L008_R2_001.fastq\n",
      "Approx 90% complete for TORN_Pool_6_S6_L008_R2_001.fastq\n",
      "Approx 95% complete for TORN_Pool_6_S6_L008_R2_001.fastq\n",
      "Analysis complete for TORN_Pool_6_S6_L008_R2_001.fastq\n",
      "Started analysis of TORN_Pool_7_S7_L006_R1_001.fastq\n",
      "Approx 5% complete for TORN_Pool_7_S7_L006_R1_001.fastq\n",
      "Approx 10% complete for TORN_Pool_7_S7_L006_R1_001.fastq\n",
      "Approx 15% complete for TORN_Pool_7_S7_L006_R1_001.fastq\n",
      "Approx 20% complete for TORN_Pool_7_S7_L006_R1_001.fastq\n",
      "Approx 25% complete for TORN_Pool_7_S7_L006_R1_001.fastq\n",
      "Approx 30% complete for TORN_Pool_7_S7_L006_R1_001.fastq\n",
      "Approx 35% complete for TORN_Pool_7_S7_L006_R1_001.fastq\n",
      "Approx 40% complete for TORN_Pool_7_S7_L006_R1_001.fastq\n",
      "Approx 45% complete for TORN_Pool_7_S7_L006_R1_001.fastq\n",
      "Approx 50% complete for TORN_Pool_7_S7_L006_R1_001.fastq\n",
      "Approx 55% complete for TORN_Pool_7_S7_L006_R1_001.fastq\n",
      "Approx 60% complete for TORN_Pool_7_S7_L006_R1_001.fastq\n",
      "Approx 65% complete for TORN_Pool_7_S7_L006_R1_001.fastq\n",
      "Approx 70% complete for TORN_Pool_7_S7_L006_R1_001.fastq\n",
      "Approx 75% complete for TORN_Pool_7_S7_L006_R1_001.fastq\n",
      "Approx 80% complete for TORN_Pool_7_S7_L006_R1_001.fastq\n",
      "Approx 85% complete for TORN_Pool_7_S7_L006_R1_001.fastq\n",
      "Approx 90% complete for TORN_Pool_7_S7_L006_R1_001.fastq\n",
      "Approx 95% complete for TORN_Pool_7_S7_L006_R1_001.fastq\n",
      "Analysis complete for TORN_Pool_7_S7_L006_R1_001.fastq\n",
      "Started analysis of TORN_Pool_7_S7_L006_R2_001.fastq\n",
      "Approx 5% complete for TORN_Pool_7_S7_L006_R2_001.fastq\n",
      "Approx 10% complete for TORN_Pool_7_S7_L006_R2_001.fastq\n",
      "Approx 15% complete for TORN_Pool_7_S7_L006_R2_001.fastq\n",
      "Approx 20% complete for TORN_Pool_7_S7_L006_R2_001.fastq\n",
      "Approx 25% complete for TORN_Pool_7_S7_L006_R2_001.fastq\n",
      "Approx 30% complete for TORN_Pool_7_S7_L006_R2_001.fastq\n",
      "Approx 35% complete for TORN_Pool_7_S7_L006_R2_001.fastq\n",
      "Approx 40% complete for TORN_Pool_7_S7_L006_R2_001.fastq\n",
      "Approx 45% complete for TORN_Pool_7_S7_L006_R2_001.fastq\n",
      "Approx 50% complete for TORN_Pool_7_S7_L006_R2_001.fastq\n",
      "Approx 55% complete for TORN_Pool_7_S7_L006_R2_001.fastq\n",
      "Approx 60% complete for TORN_Pool_7_S7_L006_R2_001.fastq\n",
      "Approx 65% complete for TORN_Pool_7_S7_L006_R2_001.fastq\n",
      "Approx 70% complete for TORN_Pool_7_S7_L006_R2_001.fastq\n",
      "Approx 75% complete for TORN_Pool_7_S7_L006_R2_001.fastq\n",
      "Approx 80% complete for TORN_Pool_7_S7_L006_R2_001.fastq\n",
      "Approx 85% complete for TORN_Pool_7_S7_L006_R2_001.fastq\n",
      "Approx 90% complete for TORN_Pool_7_S7_L006_R2_001.fastq\n",
      "Approx 95% complete for TORN_Pool_7_S7_L006_R2_001.fastq\n",
      "Analysis complete for TORN_Pool_7_S7_L006_R2_001.fastq\n",
      "Started analysis of TORN_Pool_7_S7_L008_R1_001.fastq\n",
      "Approx 5% complete for TORN_Pool_7_S7_L008_R1_001.fastq\n",
      "Approx 10% complete for TORN_Pool_7_S7_L008_R1_001.fastq\n",
      "Approx 15% complete for TORN_Pool_7_S7_L008_R1_001.fastq\n",
      "Approx 20% complete for TORN_Pool_7_S7_L008_R1_001.fastq\n",
      "Approx 25% complete for TORN_Pool_7_S7_L008_R1_001.fastq\n",
      "Approx 30% complete for TORN_Pool_7_S7_L008_R1_001.fastq\n",
      "Approx 35% complete for TORN_Pool_7_S7_L008_R1_001.fastq\n",
      "Approx 40% complete for TORN_Pool_7_S7_L008_R1_001.fastq\n",
      "Approx 45% complete for TORN_Pool_7_S7_L008_R1_001.fastq\n",
      "Approx 50% complete for TORN_Pool_7_S7_L008_R1_001.fastq\n",
      "Approx 55% complete for TORN_Pool_7_S7_L008_R1_001.fastq\n",
      "Approx 60% complete for TORN_Pool_7_S7_L008_R1_001.fastq\n",
      "Approx 65% complete for TORN_Pool_7_S7_L008_R1_001.fastq\n",
      "Approx 70% complete for TORN_Pool_7_S7_L008_R1_001.fastq\n",
      "Approx 75% complete for TORN_Pool_7_S7_L008_R1_001.fastq\n",
      "Approx 80% complete for TORN_Pool_7_S7_L008_R1_001.fastq\n",
      "Approx 85% complete for TORN_Pool_7_S7_L008_R1_001.fastq\n",
      "Approx 90% complete for TORN_Pool_7_S7_L008_R1_001.fastq\n",
      "Approx 95% complete for TORN_Pool_7_S7_L008_R1_001.fastq\n",
      "Analysis complete for TORN_Pool_7_S7_L008_R1_001.fastq\n",
      "Started analysis of TORN_Pool_7_S7_L008_R2_001.fastq\n",
      "Approx 5% complete for TORN_Pool_7_S7_L008_R2_001.fastq\n",
      "Approx 10% complete for TORN_Pool_7_S7_L008_R2_001.fastq\n",
      "Approx 15% complete for TORN_Pool_7_S7_L008_R2_001.fastq\n",
      "Approx 20% complete for TORN_Pool_7_S7_L008_R2_001.fastq\n",
      "Approx 25% complete for TORN_Pool_7_S7_L008_R2_001.fastq\n",
      "Approx 30% complete for TORN_Pool_7_S7_L008_R2_001.fastq\n",
      "Approx 35% complete for TORN_Pool_7_S7_L008_R2_001.fastq\n",
      "Approx 40% complete for TORN_Pool_7_S7_L008_R2_001.fastq\n",
      "Approx 45% complete for TORN_Pool_7_S7_L008_R2_001.fastq\n",
      "Approx 50% complete for TORN_Pool_7_S7_L008_R2_001.fastq\n",
      "Approx 55% complete for TORN_Pool_7_S7_L008_R2_001.fastq\n",
      "Approx 60% complete for TORN_Pool_7_S7_L008_R2_001.fastq\n",
      "Approx 65% complete for TORN_Pool_7_S7_L008_R2_001.fastq\n",
      "Approx 70% complete for TORN_Pool_7_S7_L008_R2_001.fastq\n",
      "Approx 75% complete for TORN_Pool_7_S7_L008_R2_001.fastq\n",
      "Approx 80% complete for TORN_Pool_7_S7_L008_R2_001.fastq\n",
      "Approx 85% complete for TORN_Pool_7_S7_L008_R2_001.fastq\n",
      "Approx 90% complete for TORN_Pool_7_S7_L008_R2_001.fastq\n",
      "Approx 95% complete for TORN_Pool_7_S7_L008_R2_001.fastq\n",
      "Analysis complete for TORN_Pool_7_S7_L008_R2_001.fastq\n",
      "Started analysis of TORN_Pool_8_S8_L006_R1_001.fastq\n",
      "Approx 5% complete for TORN_Pool_8_S8_L006_R1_001.fastq\n",
      "Approx 10% complete for TORN_Pool_8_S8_L006_R1_001.fastq\n",
      "Approx 15% complete for TORN_Pool_8_S8_L006_R1_001.fastq\n",
      "Approx 20% complete for TORN_Pool_8_S8_L006_R1_001.fastq\n",
      "Approx 25% complete for TORN_Pool_8_S8_L006_R1_001.fastq\n",
      "Approx 30% complete for TORN_Pool_8_S8_L006_R1_001.fastq\n",
      "Approx 35% complete for TORN_Pool_8_S8_L006_R1_001.fastq\n",
      "Approx 40% complete for TORN_Pool_8_S8_L006_R1_001.fastq\n",
      "Approx 45% complete for TORN_Pool_8_S8_L006_R1_001.fastq\n",
      "Approx 50% complete for TORN_Pool_8_S8_L006_R1_001.fastq\n",
      "Approx 55% complete for TORN_Pool_8_S8_L006_R1_001.fastq\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Approx 60% complete for TORN_Pool_8_S8_L006_R1_001.fastq\n",
      "Approx 65% complete for TORN_Pool_8_S8_L006_R1_001.fastq\n",
      "Approx 70% complete for TORN_Pool_8_S8_L006_R1_001.fastq\n",
      "Approx 75% complete for TORN_Pool_8_S8_L006_R1_001.fastq\n",
      "Approx 80% complete for TORN_Pool_8_S8_L006_R1_001.fastq\n",
      "Approx 85% complete for TORN_Pool_8_S8_L006_R1_001.fastq\n",
      "Approx 90% complete for TORN_Pool_8_S8_L006_R1_001.fastq\n",
      "Approx 95% complete for TORN_Pool_8_S8_L006_R1_001.fastq\n",
      "Analysis complete for TORN_Pool_8_S8_L006_R1_001.fastq\n",
      "Started analysis of TORN_Pool_8_S8_L006_R2_001.fastq\n",
      "Approx 5% complete for TORN_Pool_8_S8_L006_R2_001.fastq\n",
      "Approx 10% complete for TORN_Pool_8_S8_L006_R2_001.fastq\n",
      "Approx 15% complete for TORN_Pool_8_S8_L006_R2_001.fastq\n",
      "Approx 20% complete for TORN_Pool_8_S8_L006_R2_001.fastq\n",
      "Approx 25% complete for TORN_Pool_8_S8_L006_R2_001.fastq\n",
      "Approx 30% complete for TORN_Pool_8_S8_L006_R2_001.fastq\n",
      "Approx 35% complete for TORN_Pool_8_S8_L006_R2_001.fastq\n",
      "Approx 40% complete for TORN_Pool_8_S8_L006_R2_001.fastq\n",
      "Approx 45% complete for TORN_Pool_8_S8_L006_R2_001.fastq\n",
      "Approx 50% complete for TORN_Pool_8_S8_L006_R2_001.fastq\n",
      "Approx 55% complete for TORN_Pool_8_S8_L006_R2_001.fastq\n",
      "Approx 60% complete for TORN_Pool_8_S8_L006_R2_001.fastq\n",
      "Approx 65% complete for TORN_Pool_8_S8_L006_R2_001.fastq\n",
      "Approx 70% complete for TORN_Pool_8_S8_L006_R2_001.fastq\n",
      "Approx 75% complete for TORN_Pool_8_S8_L006_R2_001.fastq\n",
      "Approx 80% complete for TORN_Pool_8_S8_L006_R2_001.fastq\n",
      "Approx 85% complete for TORN_Pool_8_S8_L006_R2_001.fastq\n",
      "Approx 90% complete for TORN_Pool_8_S8_L006_R2_001.fastq\n",
      "Approx 95% complete for TORN_Pool_8_S8_L006_R2_001.fastq\n",
      "Analysis complete for TORN_Pool_8_S8_L006_R2_001.fastq\n",
      "Started analysis of TORN_Pool_8_S8_L008_R1_001.fastq\n",
      "Approx 5% complete for TORN_Pool_8_S8_L008_R1_001.fastq\n",
      "Approx 10% complete for TORN_Pool_8_S8_L008_R1_001.fastq\n",
      "Approx 15% complete for TORN_Pool_8_S8_L008_R1_001.fastq\n",
      "Approx 20% complete for TORN_Pool_8_S8_L008_R1_001.fastq\n",
      "Approx 25% complete for TORN_Pool_8_S8_L008_R1_001.fastq\n",
      "Approx 30% complete for TORN_Pool_8_S8_L008_R1_001.fastq\n",
      "Approx 35% complete for TORN_Pool_8_S8_L008_R1_001.fastq\n",
      "Approx 40% complete for TORN_Pool_8_S8_L008_R1_001.fastq\n",
      "Approx 45% complete for TORN_Pool_8_S8_L008_R1_001.fastq\n",
      "Approx 50% complete for TORN_Pool_8_S8_L008_R1_001.fastq\n",
      "Approx 55% complete for TORN_Pool_8_S8_L008_R1_001.fastq\n",
      "Approx 60% complete for TORN_Pool_8_S8_L008_R1_001.fastq\n",
      "Approx 65% complete for TORN_Pool_8_S8_L008_R1_001.fastq\n",
      "Approx 70% complete for TORN_Pool_8_S8_L008_R1_001.fastq\n",
      "Approx 75% complete for TORN_Pool_8_S8_L008_R1_001.fastq\n",
      "Approx 80% complete for TORN_Pool_8_S8_L008_R1_001.fastq\n",
      "Approx 85% complete for TORN_Pool_8_S8_L008_R1_001.fastq\n",
      "Approx 90% complete for TORN_Pool_8_S8_L008_R1_001.fastq\n",
      "Approx 95% complete for TORN_Pool_8_S8_L008_R1_001.fastq\n",
      "Analysis complete for TORN_Pool_8_S8_L008_R1_001.fastq\n",
      "Started analysis of TORN_Pool_8_S8_L008_R2_001.fastq\n",
      "Approx 5% complete for TORN_Pool_8_S8_L008_R2_001.fastq\n",
      "Approx 10% complete for TORN_Pool_8_S8_L008_R2_001.fastq\n",
      "Approx 15% complete for TORN_Pool_8_S8_L008_R2_001.fastq\n",
      "Approx 20% complete for TORN_Pool_8_S8_L008_R2_001.fastq\n",
      "Approx 25% complete for TORN_Pool_8_S8_L008_R2_001.fastq\n",
      "Approx 30% complete for TORN_Pool_8_S8_L008_R2_001.fastq\n",
      "Approx 35% complete for TORN_Pool_8_S8_L008_R2_001.fastq\n",
      "Approx 40% complete for TORN_Pool_8_S8_L008_R2_001.fastq\n",
      "Approx 45% complete for TORN_Pool_8_S8_L008_R2_001.fastq\n",
      "Approx 50% complete for TORN_Pool_8_S8_L008_R2_001.fastq\n",
      "Approx 55% complete for TORN_Pool_8_S8_L008_R2_001.fastq\n",
      "Approx 60% complete for TORN_Pool_8_S8_L008_R2_001.fastq\n",
      "Approx 65% complete for TORN_Pool_8_S8_L008_R2_001.fastq\n",
      "Approx 70% complete for TORN_Pool_8_S8_L008_R2_001.fastq\n",
      "Approx 75% complete for TORN_Pool_8_S8_L008_R2_001.fastq\n",
      "Approx 80% complete for TORN_Pool_8_S8_L008_R2_001.fastq\n",
      "Approx 85% complete for TORN_Pool_8_S8_L008_R2_001.fastq\n",
      "Approx 90% complete for TORN_Pool_8_S8_L008_R2_001.fastq\n",
      "Approx 95% complete for TORN_Pool_8_S8_L008_R2_001.fastq\n",
      "Analysis complete for TORN_Pool_8_S8_L008_R2_001.fastq\n",
      "Started analysis of TORN_Pool_9_S9_L006_R1_001.fastq\n",
      "Approx 5% complete for TORN_Pool_9_S9_L006_R1_001.fastq\n",
      "Approx 10% complete for TORN_Pool_9_S9_L006_R1_001.fastq\n",
      "Approx 15% complete for TORN_Pool_9_S9_L006_R1_001.fastq\n",
      "Approx 20% complete for TORN_Pool_9_S9_L006_R1_001.fastq\n",
      "Approx 25% complete for TORN_Pool_9_S9_L006_R1_001.fastq\n",
      "Approx 30% complete for TORN_Pool_9_S9_L006_R1_001.fastq\n",
      "Approx 35% complete for TORN_Pool_9_S9_L006_R1_001.fastq\n",
      "Approx 40% complete for TORN_Pool_9_S9_L006_R1_001.fastq\n",
      "Approx 45% complete for TORN_Pool_9_S9_L006_R1_001.fastq\n",
      "Approx 50% complete for TORN_Pool_9_S9_L006_R1_001.fastq\n",
      "Approx 55% complete for TORN_Pool_9_S9_L006_R1_001.fastq\n",
      "Approx 60% complete for TORN_Pool_9_S9_L006_R1_001.fastq\n",
      "Approx 65% complete for TORN_Pool_9_S9_L006_R1_001.fastq\n",
      "Approx 70% complete for TORN_Pool_9_S9_L006_R1_001.fastq\n",
      "Approx 75% complete for TORN_Pool_9_S9_L006_R1_001.fastq\n",
      "Approx 80% complete for TORN_Pool_9_S9_L006_R1_001.fastq\n",
      "Approx 85% complete for TORN_Pool_9_S9_L006_R1_001.fastq\n",
      "Approx 90% complete for TORN_Pool_9_S9_L006_R1_001.fastq\n",
      "Approx 95% complete for TORN_Pool_9_S9_L006_R1_001.fastq\n",
      "Analysis complete for TORN_Pool_9_S9_L006_R1_001.fastq\n",
      "Started analysis of TORN_Pool_9_S9_L006_R2_001.fastq\n",
      "Approx 5% complete for TORN_Pool_9_S9_L006_R2_001.fastq\n",
      "Approx 10% complete for TORN_Pool_9_S9_L006_R2_001.fastq\n",
      "Approx 15% complete for TORN_Pool_9_S9_L006_R2_001.fastq\n",
      "Approx 20% complete for TORN_Pool_9_S9_L006_R2_001.fastq\n",
      "Approx 25% complete for TORN_Pool_9_S9_L006_R2_001.fastq\n",
      "Approx 30% complete for TORN_Pool_9_S9_L006_R2_001.fastq\n",
      "Approx 35% complete for TORN_Pool_9_S9_L006_R2_001.fastq\n",
      "Approx 40% complete for TORN_Pool_9_S9_L006_R2_001.fastq\n",
      "Approx 45% complete for TORN_Pool_9_S9_L006_R2_001.fastq\n",
      "Approx 50% complete for TORN_Pool_9_S9_L006_R2_001.fastq\n",
      "Approx 55% complete for TORN_Pool_9_S9_L006_R2_001.fastq\n",
      "Approx 60% complete for TORN_Pool_9_S9_L006_R2_001.fastq\n",
      "Approx 65% complete for TORN_Pool_9_S9_L006_R2_001.fastq\n",
      "Approx 70% complete for TORN_Pool_9_S9_L006_R2_001.fastq\n",
      "Approx 75% complete for TORN_Pool_9_S9_L006_R2_001.fastq\n",
      "Approx 80% complete for TORN_Pool_9_S9_L006_R2_001.fastq\n",
      "Approx 85% complete for TORN_Pool_9_S9_L006_R2_001.fastq\n",
      "Approx 90% complete for TORN_Pool_9_S9_L006_R2_001.fastq\n",
      "Approx 95% complete for TORN_Pool_9_S9_L006_R2_001.fastq\n",
      "Analysis complete for TORN_Pool_9_S9_L006_R2_001.fastq\n",
      "Started analysis of TORN_Pool_9_S9_L008_R1_001.fastq\n",
      "Approx 5% complete for TORN_Pool_9_S9_L008_R1_001.fastq\n",
      "Approx 10% complete for TORN_Pool_9_S9_L008_R1_001.fastq\n",
      "Approx 15% complete for TORN_Pool_9_S9_L008_R1_001.fastq\n",
      "Approx 20% complete for TORN_Pool_9_S9_L008_R1_001.fastq\n",
      "Approx 25% complete for TORN_Pool_9_S9_L008_R1_001.fastq\n",
      "Approx 30% complete for TORN_Pool_9_S9_L008_R1_001.fastq\n",
      "Approx 35% complete for TORN_Pool_9_S9_L008_R1_001.fastq\n",
      "Approx 40% complete for TORN_Pool_9_S9_L008_R1_001.fastq\n",
      "Approx 45% complete for TORN_Pool_9_S9_L008_R1_001.fastq\n",
      "Approx 50% complete for TORN_Pool_9_S9_L008_R1_001.fastq\n",
      "Approx 55% complete for TORN_Pool_9_S9_L008_R1_001.fastq\n",
      "Approx 60% complete for TORN_Pool_9_S9_L008_R1_001.fastq\n",
      "Approx 65% complete for TORN_Pool_9_S9_L008_R1_001.fastq\n",
      "Approx 70% complete for TORN_Pool_9_S9_L008_R1_001.fastq\n",
      "Approx 75% complete for TORN_Pool_9_S9_L008_R1_001.fastq\n",
      "Approx 80% complete for TORN_Pool_9_S9_L008_R1_001.fastq\n",
      "Approx 85% complete for TORN_Pool_9_S9_L008_R1_001.fastq\n",
      "Approx 90% complete for TORN_Pool_9_S9_L008_R1_001.fastq\n",
      "Approx 95% complete for TORN_Pool_9_S9_L008_R1_001.fastq\n",
      "Analysis complete for TORN_Pool_9_S9_L008_R1_001.fastq\n",
      "Started analysis of TORN_Pool_9_S9_L008_R2_001.fastq\n",
      "Approx 5% complete for TORN_Pool_9_S9_L008_R2_001.fastq\n",
      "Approx 10% complete for TORN_Pool_9_S9_L008_R2_001.fastq\n",
      "Approx 15% complete for TORN_Pool_9_S9_L008_R2_001.fastq\n",
      "Approx 20% complete for TORN_Pool_9_S9_L008_R2_001.fastq\n",
      "Approx 25% complete for TORN_Pool_9_S9_L008_R2_001.fastq\n",
      "Approx 30% complete for TORN_Pool_9_S9_L008_R2_001.fastq\n",
      "Approx 35% complete for TORN_Pool_9_S9_L008_R2_001.fastq\n",
      "Approx 40% complete for TORN_Pool_9_S9_L008_R2_001.fastq\n",
      "Approx 45% complete for TORN_Pool_9_S9_L008_R2_001.fastq\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Approx 50% complete for TORN_Pool_9_S9_L008_R2_001.fastq\n",
      "Approx 55% complete for TORN_Pool_9_S9_L008_R2_001.fastq\n",
      "Approx 60% complete for TORN_Pool_9_S9_L008_R2_001.fastq\n",
      "Approx 65% complete for TORN_Pool_9_S9_L008_R2_001.fastq\n",
      "Approx 70% complete for TORN_Pool_9_S9_L008_R2_001.fastq\n",
      "Approx 75% complete for TORN_Pool_9_S9_L008_R2_001.fastq\n",
      "Approx 80% complete for TORN_Pool_9_S9_L008_R2_001.fastq\n",
      "Approx 85% complete for TORN_Pool_9_S9_L008_R2_001.fastq\n",
      "Approx 90% complete for TORN_Pool_9_S9_L008_R2_001.fastq\n",
      "Approx 95% complete for TORN_Pool_9_S9_L008_R2_001.fastq\n",
      "Analysis complete for TORN_Pool_9_S9_L008_R2_001.fastq\n"
     ]
    }
   ],
   "source": [
    "!{biotools}FastQC/fastqc \\\n",
    "{project}data/*.fastq \\\n",
    "-o {project}analysis/fastqc"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now we have .html files which can be opened in a browser to examine plots.\n",
    "\n",
    "Alternatively...  \n",
    "If on a Mac, we can download a DMG for a GUI application (see `installation.ipynb` for details). Once downloaded, open the application and select file(s) one by one. This method is slow, so the above method is prefered. "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 3. Merge files from multiple lanes\n",
    "Objective: Merge the data on lane6 and lane8 together  .\n",
    "\n",
    "Input: source files (.fastq)\n",
    "\n",
    "    {project}/data/*_L006_*.fastq      (eg. TORN_Pool_4_S4_L006_R1_001.fastqc)\n",
    "    {project}/data/*_L008_*.fastq      (eg. TORN_Pool_4_S4_L008_R1_001.fastqc)\n",
    "Output: merged files (.fastq)\n",
    "\n",
    "    {project}/analysis/merge/*.fastq   (eg. TORN_Pool_4_S4_R1.fastq)\n",
    "\n",
    "Let's do a test run on one set of files."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [],
   "source": [
    "!cat \\\n",
    "{project}data/TORN_Pool_4_S4_L006_R1_001.fastq \\\n",
    "{project}data/TORN_Pool_4_S4_L008_R1_001.fastq \\\n",
    "> {project}analysis/merge/TORN_Pool_4_S4_R1.fastq"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now let's do the rest using a loop (one for R1 and one of R2). For this I navigated to the `data/` folder in Terminal and used the following lines of code because I couldn't figure out how to use absolute paths inside a for loop in Jupyter. The output files went into `data/` but I moved them into `analysis/merge/`.\n",
    "\n",
    "R1:\n",
    "\n",
    "    (for i in *_L006_R1_001.fastq; do cat ${i%_L006_R1_001.fastq}_L006_R1_001.fastq ${i%_L006_R1_001.fastq}_L008_R1_001.fastq > ${i%_L006_R1_001.fastq}_R1.fastq; done)\n",
    "R2:\n",
    "\n",
    "    (for i in *_L006_R2_001.fastq; do cat ${i%_L006_R2_001.fastq}_L006_R2_001.fastq ${i%_L006_R2_001.fastq}_L008_R2_001.fastq > ${i%_L006_R2_001.fastq}_R2.fastq; done)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 4. Remove adapter sequences and low quality score reads (TrimGalore)\n",
    "Objective: Trim the adapter and low quality reads in the merged .fastq file.\n",
    "\n",
    "Input: merged files (.fastq)\n",
    "\n",
    "    {project}/analysis/merge/*.fastq             (eg. TORN_Pool_4_S4_R1.fastq)\n",
    "Output: trimmed files (.fq)\n",
    "\n",
    "    {project}/analysis/trimgalore/*.fastq        (eg. TORN_Pool_4_S4_R1_val_1.fq)\n",
    "    {project}/analysis/trimgalore/*.fastq        (eg. TORN_Pool_4_S4_R1.fastq_trimming_report.txt)\n",
    "    {project}/analysis/trimgalore/trim.log.txt\n",
    "Requires:\n",
    "- cutadapt (See `notebooks/installation.ipynb` for installation instructions.)\n",
    "- trim_galore (See `notebooks/installation.ipynb` for installation instructions.)\n",
    "- adapter sequences:\n",
    "    - a  = AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC\n",
    "    - a2 = AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT\n",
    "\n",
    "#### Verify cutadapt and trim_galore"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1.18\r\n"
     ]
    }
   ],
   "source": [
    "!{biotools}cutadapt-master/cutadapt --version"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\r\n",
      "                        Quality-/Adapter-/RRBS-/Hard-Trimming\r\n",
      "                                (powered by Cutadapt)\r\n",
      "                                  version 0.5.0\r\n",
      "\r\n",
      "                               Last update: 28 06 2018\r\n",
      "\r\n"
     ]
    }
   ],
   "source": [
    "!{biotools}TrimGalore-0.5.0/trim_galore --version"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Both are installed and up-to-date.\n",
    "#### Run TrimGalore\n",
    "\n",
    "I tried to run TrimGalore on one sample, but couldn't get to work. `(cutadapt: error: Too many parameters.)`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "Path to Cutadapt set as: '/Applications/bio-tools/cutadapt-master/cutadapt' (user defined)\n",
      "1.18\n",
      "Cutadapt seems to be working fine (tested command '/Applications/bio-tools/cutadapt-master/cutadapt --version')\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> /Users/calderatta/Desktop/FISH 546 - Bioinformatics/project/data/TORN_Pool_10_S10_L008_R2_001.fastq <<)\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "Illumina\t62856\tAGATCGGAAGAGC\t1000000\t6.29\n",
      "Nextera\t6\tCTGTCTCTTATA\t1000000\t0.00\n",
      "smallRNA\t1\tTGGAATTCTCGG\t1000000\t0.00\n",
      "Using Illumina adapter for trimming (count: 62856). Second best hit was Nextera (count: 6)\n",
      "\n",
      "Writing report to '/Users/calderatta/Desktop/FISH 546 - Bioinformatics/project/analysis/trimgalore/TORN_Pool_10_S10_L008_R2_001.fastq_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: /Users/calderatta/Desktop/FISH 546 - Bioinformatics/project/data/TORN_Pool_10_S10_L008_R2_001.fastq\n",
      "Trimming mode: single-end\n",
      "Trim Galore version: 0.5.0\n",
      "Cutadapt version: 1.18\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length before a sequence gets removed: 20 bp\n",
      "\n",
      "Writing final adapter and quality trimmed output to TORN_Pool_10_S10_L008_R2_001_trimmed.fq\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff 20) and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file /Users/calderatta/Desktop/FISH 546 - Bioinformatics/project/data/TORN_Pool_10_S10_L008_R2_001.fastq <<< \n",
      "Run \"cutadapt --help\" to see command-line options.\n",
      "See https://cutadapt.readthedocs.io/ for full documentation.\n",
      "\n",
      "cutadapt: error: Too many parameters.\n",
      "\n",
      "\n",
      "Cutadapt terminated with exit signal: '512'.\n",
      "Terminating Trim Galore run, please check error message(s) to get an idea what went wrong...\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!{biotools}TrimGalore-0.5.0/trim_galore \\\n",
    "{project}data/TORN_Pool_10_S10_L008_R2_001.fastq \\\n",
    "--path_to_cutadapt {biotools}cutadapt-master/cutadapt \\\n",
    "-o {project}analysis/trimgalore/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "As with merge, I also navigated to the `analysis/merge/` folder in Terminal and used the following line of code. Each file took several minutes to complete. The output files went into `analysis/merge/` but I moved them into `analysis/trimgalore/`.\n",
    "\n",
    "    (for i in *_R1.fastq; do /Applications/bio-tools/TrimGalore-0.5.0/trim_galore --path_to_cutadapt /Applications/bio-tools/cutadapt-master/cutadapt -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC -a2 AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT --paired ${i%_R1.fastq}_R1.fastq ${i%_R1.fastq}_R2.fastq; done) >& trim.log.txt\n",
    "#### Rename\n",
    "The output file contains the convention 'val_1' (eg. `TORN_Pool_4_S4_R1_val_1.fq`) or 'val_2', which need to be removed.  \n",
    "Note: There is also a script containing the following code: `{project}kuang-et-al-2018/scripts-old/preads_script/rename.sh`."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "!for f in *_val_1.fq; do /\n",
    "mv -- \"$f\" \"${f%_R1_val_1.fq}_R1.fq\"; /\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "!for f in *_val_2.fq; do /\n",
    "mv -- \"$f\" \"${f%_R2_val_2.fq}_R2.fq\"; /\n",
    "done)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 5. Remove the duplicates from PCR, parse the reads to each locus (preads)\n",
    "Preads is a custom perl script for removing duplicate reads.\n",
    "\n",
    "Objective: Remove replicates in trimmed file, and parse them into corresponding genes.\n",
    "\n",
    "Input:  all *.fq files, baits sequences,rmrep.pl, bandp.pl, data.pm\n",
    "\n",
    "    {project}/analysis/trimgalore/*_R1.fq (eg. TORN_Pool_4_S4_R1.fq)\n",
    "    {project}/analysis/trimgalore/*_R2.fq (eg. TORN_Pool_4_S4_R2.fq)\n",
    "    {project}kuang-et-al-2018/scripts-old/preads_script/MudSkipper_Kit_Plus_Opsins.fas\n",
    "    {project}kuang-et-al-2018/scripts-old/preads_script/bandp.pl\n",
    "    {project}kuang-et-al-2018/scripts-old/preads_script/data.pm\n",
    "    {project}kuang-et-al-2018/scripts-old/preads_script/rmrep.pl\n",
    "Output (rmrep.pl): gene files containing corresponding reads(.fq)\n",
    "\n",
    "    {project}/analysis/preads/*_rmrep_R1.fq      (eg. TORN_Pool_4_S4_rmrep_R1.fq)\n",
    "    {project}/analysis/preads/*_rmrep_R1.index   (eg. TORN_Pool_4_S4_rmrep_R1.index)\n",
    "    {project}/analysis/preads/*_rmrep_R2.fq      (eg. TORN_Pool_4_S4_rmrep_R2.fq)\n",
    "    {project}/analysis/preads/*_rmrep_R2.index   (eg. TORN_Pool_4_S4_rmrep_R2.index)\n",
    "    {project}/analysis/preads/*.fas              (eg. TORN_Pool_4_S4.fas)\n",
    "    {project}/analysis/preads/*.index            (eg. TORN_Pool_4_S4.index)\n",
    "    {project}/analysis/preads/*.nhr              (eg. TORN_Pool_4_S4.nhr)\n",
    "    {project}/analysis/preads/*.nin              (eg. TORN_Pool_4_S4.nin)\n",
    "    {project}/analysis/preads/*.nsq              (eg. TORN_Pool_4_S4.nsq)\n",
    "    {project}/analysis/preads/preads.1.log.txt\n",
    "Output (bandp.pl): gene files containing corresponding reads(.fq)\n",
    "\n",
    "    {project}/analysis/preads/*.*.blast.txt      (eg. MudSkipper_Kit_Plus_Opsins.TORN_Pool_4_S4.blast.txt)\n",
    "    {project}/analysis/preads/*_results/         (eg. TORN_Pool_4_S4_results) <- contains many .fq files\n",
    "    {project}/analysis/preads/preads.1.log.txt\n",
    "        \n",
    "Requires:\n",
    "- makeblastdb (See `notebooks/installation.ipynb` for installation instructions. Make sure the script is added to a PATH directory. Eg. `/usr/local/bin/`)\n",
    "- blastn (See `notebooks/installation.ipynb` for installation instructions. Make sure the script is added to a PATH directory. Eg. `/usr/local/bin/`)\n",
    "\n",
    "Usage example:\n",
    "\n",
    "    ./rmrep.pl -taxalist=\"558_1\"\n",
    "    ./bandp.pl -query=\"Python_molurus\" -subject=\"558_1\" > preads.1.log.txt\n",
    "#### Setup\n",
    "First we need to move all the trim_galore results (.fq) into the same directory as the other input files. In Terminal:\n",
    "\n",
    "    cd /Users/calderatta/Desktop/FISH\\ 546\\ -\\ Bioinformatics/project/kuang-et-al-2018/scripts-old/preads_script/ \n",
    "#### Run rmrep.pl\n",
    "This will be a long-running job (~15min per sample), so let's test it on one first. In the actuall run, we want to substitute '558_1' in the example with a list of sample names (a string of names separated by spaces). For the test we will use 'TORN_Pool_4_S4'.\n",
    "\n",
    "    ./rmrep.pl -taxalist=\"TORN_Pool_4_S4\"\n",
    "Ok. It works. Now we can run it on a for loop.\n",
    "\n",
    "    (for i in TORN_Pool_4_S4 TORN_Pool_5_S5 TORN_Pool_6_S6 TORN_Pool_7_S7 TORN_Pool_8_S8 TORN_Pool_9_S9 TORN_Pool_10_S10; do ./rmrep.pl -taxalist=$i; done)\n",
    "#### Run bandp.pl\n",
    "\n",
    "    ./bandp.pl -query=\"MudSkipper_Kit_Plus_Opsins\" -subject=TORN_Pool_4_S4 > preads.1.log.txt\n",
    "Inputting the subject argument as a list of names separated by spaces didn't seem to work so for for this I just did them individually. Each job took between 24 and 48 hours and done on Office Desktop Windows, sample 9 and 10 done on MOX.\n",
    "\n",
    "For files with the prefix 'CG_LWS_', the output on Windows inserts a '|' in the file convention on MOX and a character similar to '•' in Windows, which gets converted to a questionmark-box character when moved to Mac.I substituted these characters with an underscore to be able to input into Trinity.\n",
    "\n",
    "    rename -n 's/|/_/' *.fq\n",
    "#### Move Files to Analysis Directory\n",
    "Move output files to the new `/analysis/preads/`."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 6. Assemble the filtered reads into contigs (Trinity)\n",
    "\n",
    "Objective: Assemble the reads to short contigs .\n",
    "\n",
    "Input: Samplename_results (preads result) folder containing gene files, runtrinity.pl, .sh file\n",
    "\n",
    "    {project}/analysis/preads/*.*.blast.txt      (eg. MudSkipper_Kit_Plus_Opsins.TORN_Pool_4_S4.blast.txt)\n",
    "    {project}/analysis/preads/*_results/         (eg. TORN_Pool_4_S4_results)\n",
    "    {project}kuang-et-al-2018/scripts-old/./runtrinity.pl\n",
    "Output (rmrep.pl): fasta files named in format of \"gene name.Trinity.fasta\".\n",
    "\n",
    "    {project}/analysis/trinity/*.Trinity.fasta\n",
    "        \n",
    "Requires:\n",
    "- Trinity (See `notebooks/installation.ipynb` for installation instructions. Make sure the script is added to a PATH directory. Eg. `/usr/local/bin/`)\n",
    "\n",
    "Usage example:\n",
    "\n",
    "    ./runtrinity.pl -species=\"samplename1 samplename2\" > runtrinity15.log\n",
    "\n",
    "#### Setup\n",
    "First we need to move all the input files, preads results and 'runtrinity.pl' (copied), into a new directory:\n",
    "\n",
    "    /Users/calderatta/Desktop/FISH\\ 546\\ -\\ Bioinformatics/project/analysis/trinity\n",
    "#### Run runtrinity.pl\n",
    "In Progress...\n",
    "\n",
    "Running Trinity requires ParaFly (in Trinity), seqtk-trinity (in Trinity), samtools, jellyfish (http://www.genome.umd.edu/jellyfish.html#Release), salmon (https://combine-lab.github.io/salmon/)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
