{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Exon-capture to Phylogeny Project\n",
    "\n",
    "Calder Atta - FISH 546 Project"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Workflow (from Kuang et al. 2018)\n",
    "\n",
    "1. Raw reads from Illumina sequencer\n",
    "2. BCL to fastq format demultiplex (Illumina bcl2fastq package)\n",
    "3. Remove adapter sequences and low quality score reads (Cutadapt 1.1 Trim_golare v0.2.8)\n",
    "4. Remove the duplicates from PCR, parse the reads to each locus (A custom perl script: preads (supplementary))\n",
    "5. Assemble the filtered reads into contigs (Trinity v20140717)\n",
    "6. Merge the loci containing more than one contigs (Geneious v7.1.5)\n",
    "7. Retrieve orthology by pairwise alignment to corresponding baits sequence (A custom Perl script: Smith-Waterman algorithm)\n",
    "8. Identify orthology by comparing the retrieved sequence to the genome of O. nilotics (Blast v2.2.27)\n",
    "9. Multiple sequences alignment (Clustal Omega v1.1.1)\n",
    "10. Downstream analysis"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Notes from Kuang et al. 2018\n",
    "\n",
    "- Samples and Genes\n",
    "\t- Sampled 43 species\n",
    "\t- 1 mt markers (COI)\n",
    "\t- 17817 nu markers (120bp baits)\n",
    "\t\t- target region <120bp was padded with T to 120bp\n",
    "- Filtering\n",
    "\t- only used sequences found in all species and <5% missing data -> 570 markers\n",
    "\t- parameters for evaluating usefulness (calculated for all markers)\n",
    "\t\t1. Average pairwise difference (p-dist)\n",
    "\t\t2. Molecular clocklikeness (MCL)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Getting started\n",
    "Set short cuts for frequently used directories inside variables."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "project = \"/Users/calderatta/Desktop/FISH\\ 546\\ -\\ Bioinformatics/project/\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "biotools = \"/Applications/bio-tools/\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Examine raw data."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[31mTORN_Pool_10_S10_L006_R1_001.fastq\u001b[m\u001b[m* \u001b[31mTORN_Pool_6_S6_L008_R1_001.fastq\u001b[m\u001b[m*\r\n",
      "\u001b[31mTORN_Pool_10_S10_L006_R2_001.fastq\u001b[m\u001b[m* \u001b[31mTORN_Pool_6_S6_L008_R2_001.fastq\u001b[m\u001b[m*\r\n",
      "\u001b[31mTORN_Pool_10_S10_L008_R1_001.fastq\u001b[m\u001b[m* \u001b[31mTORN_Pool_7_S7_L006_R1_001.fastq\u001b[m\u001b[m*\r\n",
      "\u001b[31mTORN_Pool_10_S10_L008_R2_001.fastq\u001b[m\u001b[m* \u001b[31mTORN_Pool_7_S7_L006_R2_001.fastq\u001b[m\u001b[m*\r\n",
      "\u001b[31mTORN_Pool_4_S4_L006_R1_001.fastq\u001b[m\u001b[m*   \u001b[31mTORN_Pool_7_S7_L008_R1_001.fastq\u001b[m\u001b[m*\r\n",
      "\u001b[31mTORN_Pool_4_S4_L006_R2_001.fastq\u001b[m\u001b[m*   \u001b[31mTORN_Pool_7_S7_L008_R2_001.fastq\u001b[m\u001b[m*\r\n",
      "\u001b[31mTORN_Pool_4_S4_L008_R1_001.fastq\u001b[m\u001b[m*   \u001b[31mTORN_Pool_8_S8_L006_R1_001.fastq\u001b[m\u001b[m*\r\n",
      "\u001b[31mTORN_Pool_4_S4_L008_R2_001.fastq\u001b[m\u001b[m*   \u001b[31mTORN_Pool_8_S8_L006_R2_001.fastq\u001b[m\u001b[m*\r\n",
      "\u001b[31mTORN_Pool_5_S5_L006_R1_001.fastq\u001b[m\u001b[m*   \u001b[31mTORN_Pool_8_S8_L008_R1_001.fastq\u001b[m\u001b[m*\r\n",
      "\u001b[31mTORN_Pool_5_S5_L006_R2_001.fastq\u001b[m\u001b[m*   \u001b[31mTORN_Pool_8_S8_L008_R2_001.fastq\u001b[m\u001b[m*\r\n",
      "\u001b[31mTORN_Pool_5_S5_L008_R1_001.fastq\u001b[m\u001b[m*   \u001b[31mTORN_Pool_9_S9_L006_R1_001.fastq\u001b[m\u001b[m*\r\n",
      "\u001b[31mTORN_Pool_5_S5_L008_R2_001.fastq\u001b[m\u001b[m*   \u001b[31mTORN_Pool_9_S9_L006_R2_001.fastq\u001b[m\u001b[m*\r\n",
      "\u001b[31mTORN_Pool_6_S6_L006_R1_001.fastq\u001b[m\u001b[m*   \u001b[31mTORN_Pool_9_S9_L008_R1_001.fastq\u001b[m\u001b[m*\r\n",
      "\u001b[31mTORN_Pool_6_S6_L006_R2_001.fastq\u001b[m\u001b[m*   \u001b[31mTORN_Pool_9_S9_L008_R2_001.fastq\u001b[m\u001b[m*\r\n"
     ]
    }
   ],
   "source": [
    "ls {project}raw"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Species 4 through 10 should be listed. For each species there are forward (R1) and reverse (R2) reads for each Illumina lane (L00#). Within each forward/reverse file pairs, the order of sequences is consistent."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "@K00179:70:HHV7JBBXX:6:1101:24454:1209 1:N:0:AACGAAGT\r\n",
      "TNTCTCTCTCTCTTGCTCTCTCTCTCTCTGTTTGAGCTCTCTCTCCCTCTCTCTCTCTCTCTGTCTCTCTCTGTTTGAGCTAACTCTCTCTCTCTGTTAGAGCTCTCTCTCGNTCGCTCTCTCTCGCTNTCGCTGGCTCGCACGCTCTCT\r\n",
      "+\r\n",
      "A#AAAFFFJJJFJFAJFJ-FFFAJFJ-FJ-AA<--7<-7FA7F<A--77A-7A-<777A-<F--A-<FF7F<-----777777F777A-77AF<---7A-----)-7-7-7)#--7--<)F<--7)-)#7A-77)--)7)))7)7)))--\r\n",
      "@K00179:70:HHV7JBBXX:6:1101:26829:1209 1:N:0:AACGAAGT\r\n",
      "CNCTTTCCTTCAGGAGAGACTCTGTCAGGAGGTGCAGGAGGAACAAAAGGAGCAAGAGGAGGAGGATCTGAAGGAGGGATGAGGTGTTGCAGGACGATGAACAGGAGGGGGAGCATGAGGAGGAGCAGGAGTAGGTGGAGCATAAGGAGG\r\n",
      "+\r\n",
      "A#AAFFJFFJJJJJJAFJJJAJFJ<JAAJJJJAJJJ7FFF<AJJFAJFFJ<AFJJF<AJJJFFJA77F7A--F-AAFF-7FJJJAF7FJ<AAFFJJ77AF7<<A<AJF))-))-)-77A<-<7AJF)<7)7-7-7<F-))7)---7--7)\r\n",
      "@K00179:70:HHV7JBBXX:6:1101:4472:1226 1:N:0:AACGAAGT\r\n",
      "GNAACAACATGGAGGTCAGAGGAGGAACAACATGGAGGTCAGAGGAGGAACAACATGGAGGTCAGAGGAGGAACAACATGGAGGTCAGAGAAGGCGCATCACGTATCTCAGANGAAAAGAAAGGAGGTNTGCAAAGACGAACGAGGGGGC\r\n"
     ]
    }
   ],
   "source": [
    "!head {project}data/TORN_Pool_10_S10_L006_R1_001.fastq"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now let's check that the number of sequences in R1 and R2 files match. First test on one file."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2419885\r\n"
     ]
    }
   ],
   "source": [
    "!grep -c \"@\" {project}data/TORN_Pool_10_S10_L008_R2_001.fastq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/calderatta/Desktop/FISH 546 - Bioinformatics/project/raw/TORN_Pool_10_S10_L006_R1_001.fastq:2419885\n",
      "/Users/calderatta/Desktop/FISH 546 - Bioinformatics/project/raw/TORN_Pool_10_S10_L006_R2_001.fastq:2419885\n",
      "/Users/calderatta/Desktop/FISH 546 - Bioinformatics/project/raw/TORN_Pool_10_S10_L008_R1_001.fastq:1497568\n",
      "/Users/calderatta/Desktop/FISH 546 - Bioinformatics/project/raw/TORN_Pool_10_S10_L008_R2_001.fastq:1497568\n",
      "/Users/calderatta/Desktop/FISH 546 - Bioinformatics/project/raw/TORN_Pool_4_S4_L006_R1_001.fastq:3061218\n",
      "/Users/calderatta/Desktop/FISH 546 - Bioinformatics/project/raw/TORN_Pool_4_S4_L006_R2_001.fastq:3061218\n",
      "/Users/calderatta/Desktop/FISH 546 - Bioinformatics/project/raw/TORN_Pool_4_S4_L008_R1_001.fastq:2506641\n",
      "/Users/calderatta/Desktop/FISH 546 - Bioinformatics/project/raw/TORN_Pool_4_S4_L008_R2_001.fastq:2506641\n",
      "/Users/calderatta/Desktop/FISH 546 - Bioinformatics/project/raw/TORN_Pool_5_S5_L006_R1_001.fastq:1248025\n",
      "/Users/calderatta/Desktop/FISH 546 - Bioinformatics/project/raw/TORN_Pool_5_S5_L006_R2_001.fastq:1248025\n",
      "/Users/calderatta/Desktop/FISH 546 - Bioinformatics/project/raw/TORN_Pool_5_S5_L008_R1_001.fastq:870104\n",
      "/Users/calderatta/Desktop/FISH 546 - Bioinformatics/project/raw/TORN_Pool_5_S5_L008_R2_001.fastq:870104\n",
      "/Users/calderatta/Desktop/FISH 546 - Bioinformatics/project/raw/TORN_Pool_6_S6_L006_R1_001.fastq:3406559\n",
      "/Users/calderatta/Desktop/FISH 546 - Bioinformatics/project/raw/TORN_Pool_6_S6_L006_R2_001.fastq:3406559\n",
      "/Users/calderatta/Desktop/FISH 546 - Bioinformatics/project/raw/TORN_Pool_6_S6_L008_R1_001.fastq:3107867\n",
      "/Users/calderatta/Desktop/FISH 546 - Bioinformatics/project/raw/TORN_Pool_6_S6_L008_R2_001.fastq:3107867\n",
      "/Users/calderatta/Desktop/FISH 546 - Bioinformatics/project/raw/TORN_Pool_7_S7_L006_R1_001.fastq:2565256\n",
      "/Users/calderatta/Desktop/FISH 546 - Bioinformatics/project/raw/TORN_Pool_7_S7_L006_R2_001.fastq:2565256\n",
      "/Users/calderatta/Desktop/FISH 546 - Bioinformatics/project/raw/TORN_Pool_7_S7_L008_R1_001.fastq:3177384\n",
      "/Users/calderatta/Desktop/FISH 546 - Bioinformatics/project/raw/TORN_Pool_7_S7_L008_R2_001.fastq:3177384\n",
      "/Users/calderatta/Desktop/FISH 546 - Bioinformatics/project/raw/TORN_Pool_8_S8_L006_R1_001.fastq:2083571\n",
      "/Users/calderatta/Desktop/FISH 546 - Bioinformatics/project/raw/TORN_Pool_8_S8_L006_R2_001.fastq:2083571\n",
      "/Users/calderatta/Desktop/FISH 546 - Bioinformatics/project/raw/TORN_Pool_8_S8_L008_R1_001.fastq:2274644\n",
      "/Users/calderatta/Desktop/FISH 546 - Bioinformatics/project/raw/TORN_Pool_8_S8_L008_R2_001.fastq:2274644\n",
      "/Users/calderatta/Desktop/FISH 546 - Bioinformatics/project/raw/TORN_Pool_9_S9_L006_R1_001.fastq:2924224\n",
      "/Users/calderatta/Desktop/FISH 546 - Bioinformatics/project/raw/TORN_Pool_9_S9_L006_R2_001.fastq:2924224\n",
      "/Users/calderatta/Desktop/FISH 546 - Bioinformatics/project/raw/TORN_Pool_9_S9_L008_R1_001.fastq:2114997\n",
      "/Users/calderatta/Desktop/FISH 546 - Bioinformatics/project/raw/TORN_Pool_9_S9_L008_R2_001.fastq:2114997\n"
     ]
    }
   ],
   "source": [
    "!grep -c '@' {project}data/*.fastq"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 1. Raw Reads from Illumina Sequencer\n",
    "This was already done."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 2. BCL to fastq format demultiplex\n",
    "This was already done."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 3. Remove adapter sequences and low quality score reads"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### FastQC - checking quality across all squences in each file"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### From DMG (Mac GUI application)\n",
    "Open the application, and select file(s)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "!open {biotools}FastQC.app"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### From Zip File (Windows or Mac)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Started analysis of TORN_Pool_10_S10_L006_R1_001.fastq\n",
      "Approx 5% complete for TORN_Pool_10_S10_L006_R1_001.fastq\n",
      "Approx 10% complete for TORN_Pool_10_S10_L006_R1_001.fastq\n",
      "Approx 15% complete for TORN_Pool_10_S10_L006_R1_001.fastq\n",
      "Approx 20% complete for TORN_Pool_10_S10_L006_R1_001.fastq\n",
      "Approx 25% complete for TORN_Pool_10_S10_L006_R1_001.fastq\n",
      "Approx 30% complete for TORN_Pool_10_S10_L006_R1_001.fastq\n",
      "Approx 35% complete for TORN_Pool_10_S10_L006_R1_001.fastq\n",
      "Approx 40% complete for TORN_Pool_10_S10_L006_R1_001.fastq\n",
      "Approx 45% complete for TORN_Pool_10_S10_L006_R1_001.fastq\n",
      "Approx 50% complete for TORN_Pool_10_S10_L006_R1_001.fastq\n",
      "Approx 55% complete for TORN_Pool_10_S10_L006_R1_001.fastq\n",
      "Approx 60% complete for TORN_Pool_10_S10_L006_R1_001.fastq\n",
      "Approx 65% complete for TORN_Pool_10_S10_L006_R1_001.fastq\n",
      "Approx 70% complete for TORN_Pool_10_S10_L006_R1_001.fastq\n",
      "Approx 75% complete for TORN_Pool_10_S10_L006_R1_001.fastq\n",
      "Approx 80% complete for TORN_Pool_10_S10_L006_R1_001.fastq\n",
      "Approx 85% complete for TORN_Pool_10_S10_L006_R1_001.fastq\n",
      "Approx 90% complete for TORN_Pool_10_S10_L006_R1_001.fastq\n",
      "Approx 95% complete for TORN_Pool_10_S10_L006_R1_001.fastq\n",
      "Analysis complete for TORN_Pool_10_S10_L006_R1_001.fastq\n",
      "Started analysis of TORN_Pool_10_S10_L006_R2_001.fastq\n",
      "Approx 5% complete for TORN_Pool_10_S10_L006_R2_001.fastq\n",
      "Approx 10% complete for TORN_Pool_10_S10_L006_R2_001.fastq\n",
      "Approx 15% complete for TORN_Pool_10_S10_L006_R2_001.fastq\n",
      "Approx 20% complete for TORN_Pool_10_S10_L006_R2_001.fastq\n",
      "Approx 25% complete for TORN_Pool_10_S10_L006_R2_001.fastq\n",
      "Approx 30% complete for TORN_Pool_10_S10_L006_R2_001.fastq\n",
      "Approx 35% complete for TORN_Pool_10_S10_L006_R2_001.fastq\n",
      "Approx 40% complete for TORN_Pool_10_S10_L006_R2_001.fastq\n",
      "Approx 45% complete for TORN_Pool_10_S10_L006_R2_001.fastq\n",
      "Approx 50% complete for TORN_Pool_10_S10_L006_R2_001.fastq\n",
      "Approx 55% complete for TORN_Pool_10_S10_L006_R2_001.fastq\n",
      "Approx 60% complete for TORN_Pool_10_S10_L006_R2_001.fastq\n",
      "Approx 65% complete for TORN_Pool_10_S10_L006_R2_001.fastq\n",
      "Approx 70% complete for TORN_Pool_10_S10_L006_R2_001.fastq\n",
      "Approx 75% complete for TORN_Pool_10_S10_L006_R2_001.fastq\n",
      "Approx 80% complete for TORN_Pool_10_S10_L006_R2_001.fastq\n",
      "Approx 85% complete for TORN_Pool_10_S10_L006_R2_001.fastq\n",
      "Approx 90% complete for TORN_Pool_10_S10_L006_R2_001.fastq\n",
      "Approx 95% complete for TORN_Pool_10_S10_L006_R2_001.fastq\n",
      "Analysis complete for TORN_Pool_10_S10_L006_R2_001.fastq\n",
      "Started analysis of TORN_Pool_10_S10_L008_R1_001.fastq\n",
      "Approx 5% complete for TORN_Pool_10_S10_L008_R1_001.fastq\n",
      "Approx 10% complete for TORN_Pool_10_S10_L008_R1_001.fastq\n",
      "Approx 15% complete for TORN_Pool_10_S10_L008_R1_001.fastq\n",
      "Approx 20% complete for TORN_Pool_10_S10_L008_R1_001.fastq\n",
      "Approx 25% complete for TORN_Pool_10_S10_L008_R1_001.fastq\n",
      "Approx 30% complete for TORN_Pool_10_S10_L008_R1_001.fastq\n",
      "Approx 35% complete for TORN_Pool_10_S10_L008_R1_001.fastq\n",
      "Approx 40% complete for TORN_Pool_10_S10_L008_R1_001.fastq\n",
      "Approx 45% complete for TORN_Pool_10_S10_L008_R1_001.fastq\n",
      "Approx 50% complete for TORN_Pool_10_S10_L008_R1_001.fastq\n",
      "Approx 55% complete for TORN_Pool_10_S10_L008_R1_001.fastq\n",
      "Approx 60% complete for TORN_Pool_10_S10_L008_R1_001.fastq\n",
      "Approx 65% complete for TORN_Pool_10_S10_L008_R1_001.fastq\n",
      "Approx 70% complete for TORN_Pool_10_S10_L008_R1_001.fastq\n",
      "Approx 75% complete for TORN_Pool_10_S10_L008_R1_001.fastq\n",
      "Approx 80% complete for TORN_Pool_10_S10_L008_R1_001.fastq\n",
      "Approx 85% complete for TORN_Pool_10_S10_L008_R1_001.fastq\n",
      "Approx 90% complete for TORN_Pool_10_S10_L008_R1_001.fastq\n",
      "Approx 95% complete for TORN_Pool_10_S10_L008_R1_001.fastq\n",
      "Analysis complete for TORN_Pool_10_S10_L008_R1_001.fastq\n",
      "Started analysis of TORN_Pool_10_S10_L008_R2_001.fastq\n",
      "Approx 5% complete for TORN_Pool_10_S10_L008_R2_001.fastq\n",
      "Approx 10% complete for TORN_Pool_10_S10_L008_R2_001.fastq\n",
      "Approx 15% complete for TORN_Pool_10_S10_L008_R2_001.fastq\n",
      "Approx 20% complete for TORN_Pool_10_S10_L008_R2_001.fastq\n",
      "Approx 25% complete for TORN_Pool_10_S10_L008_R2_001.fastq\n",
      "Approx 30% complete for TORN_Pool_10_S10_L008_R2_001.fastq\n",
      "Approx 35% complete for TORN_Pool_10_S10_L008_R2_001.fastq\n",
      "Approx 40% complete for TORN_Pool_10_S10_L008_R2_001.fastq\n",
      "Approx 45% complete for TORN_Pool_10_S10_L008_R2_001.fastq\n",
      "Approx 50% complete for TORN_Pool_10_S10_L008_R2_001.fastq\n",
      "Approx 55% complete for TORN_Pool_10_S10_L008_R2_001.fastq\n",
      "Approx 60% complete for TORN_Pool_10_S10_L008_R2_001.fastq\n",
      "Approx 65% complete for TORN_Pool_10_S10_L008_R2_001.fastq\n",
      "Approx 70% complete for TORN_Pool_10_S10_L008_R2_001.fastq\n",
      "Approx 75% complete for TORN_Pool_10_S10_L008_R2_001.fastq\n",
      "Approx 80% complete for TORN_Pool_10_S10_L008_R2_001.fastq\n",
      "Approx 85% complete for TORN_Pool_10_S10_L008_R2_001.fastq\n",
      "Approx 90% complete for TORN_Pool_10_S10_L008_R2_001.fastq\n",
      "Approx 95% complete for TORN_Pool_10_S10_L008_R2_001.fastq\n",
      "Analysis complete for TORN_Pool_10_S10_L008_R2_001.fastq\n",
      "Started analysis of TORN_Pool_4_S4_L006_R1_001.fastq\n",
      "Approx 5% complete for TORN_Pool_4_S4_L006_R1_001.fastq\n",
      "Approx 10% complete for TORN_Pool_4_S4_L006_R1_001.fastq\n",
      "Approx 15% complete for TORN_Pool_4_S4_L006_R1_001.fastq\n",
      "Approx 20% complete for TORN_Pool_4_S4_L006_R1_001.fastq\n",
      "Approx 25% complete for TORN_Pool_4_S4_L006_R1_001.fastq\n",
      "Approx 30% complete for TORN_Pool_4_S4_L006_R1_001.fastq\n",
      "Approx 35% complete for TORN_Pool_4_S4_L006_R1_001.fastq\n",
      "Approx 40% complete for TORN_Pool_4_S4_L006_R1_001.fastq\n",
      "Approx 45% complete for TORN_Pool_4_S4_L006_R1_001.fastq\n",
      "Approx 50% complete for TORN_Pool_4_S4_L006_R1_001.fastq\n",
      "Approx 55% complete for TORN_Pool_4_S4_L006_R1_001.fastq\n",
      "Approx 60% complete for TORN_Pool_4_S4_L006_R1_001.fastq\n",
      "Approx 65% complete for TORN_Pool_4_S4_L006_R1_001.fastq\n",
      "Approx 70% complete for TORN_Pool_4_S4_L006_R1_001.fastq\n",
      "Approx 75% complete for TORN_Pool_4_S4_L006_R1_001.fastq\n",
      "Approx 80% complete for TORN_Pool_4_S4_L006_R1_001.fastq\n",
      "Approx 85% complete for TORN_Pool_4_S4_L006_R1_001.fastq\n",
      "Approx 90% complete for TORN_Pool_4_S4_L006_R1_001.fastq\n",
      "Approx 95% complete for TORN_Pool_4_S4_L006_R1_001.fastq\n",
      "Analysis complete for TORN_Pool_4_S4_L006_R1_001.fastq\n",
      "Started analysis of TORN_Pool_4_S4_L006_R2_001.fastq\n",
      "Approx 5% complete for TORN_Pool_4_S4_L006_R2_001.fastq\n",
      "Approx 10% complete for TORN_Pool_4_S4_L006_R2_001.fastq\n",
      "Approx 15% complete for TORN_Pool_4_S4_L006_R2_001.fastq\n",
      "Approx 20% complete for TORN_Pool_4_S4_L006_R2_001.fastq\n",
      "Approx 25% complete for TORN_Pool_4_S4_L006_R2_001.fastq\n",
      "Approx 30% complete for TORN_Pool_4_S4_L006_R2_001.fastq\n",
      "Approx 35% complete for TORN_Pool_4_S4_L006_R2_001.fastq\n",
      "Approx 40% complete for TORN_Pool_4_S4_L006_R2_001.fastq\n",
      "Approx 45% complete for TORN_Pool_4_S4_L006_R2_001.fastq\n",
      "Approx 50% complete for TORN_Pool_4_S4_L006_R2_001.fastq\n",
      "Approx 55% complete for TORN_Pool_4_S4_L006_R2_001.fastq\n",
      "Approx 60% complete for TORN_Pool_4_S4_L006_R2_001.fastq\n",
      "Approx 65% complete for TORN_Pool_4_S4_L006_R2_001.fastq\n",
      "Approx 70% complete for TORN_Pool_4_S4_L006_R2_001.fastq\n",
      "Approx 75% complete for TORN_Pool_4_S4_L006_R2_001.fastq\n",
      "Approx 80% complete for TORN_Pool_4_S4_L006_R2_001.fastq\n",
      "Approx 85% complete for TORN_Pool_4_S4_L006_R2_001.fastq\n",
      "Approx 90% complete for TORN_Pool_4_S4_L006_R2_001.fastq\n",
      "Approx 95% complete for TORN_Pool_4_S4_L006_R2_001.fastq\n",
      "Analysis complete for TORN_Pool_4_S4_L006_R2_001.fastq\n",
      "Started analysis of TORN_Pool_4_S4_L008_R1_001.fastq\n",
      "Approx 5% complete for TORN_Pool_4_S4_L008_R1_001.fastq\n",
      "Approx 10% complete for TORN_Pool_4_S4_L008_R1_001.fastq\n",
      "Approx 15% complete for TORN_Pool_4_S4_L008_R1_001.fastq\n",
      "Approx 20% complete for TORN_Pool_4_S4_L008_R1_001.fastq\n",
      "Approx 25% complete for TORN_Pool_4_S4_L008_R1_001.fastq\n",
      "Approx 30% complete for TORN_Pool_4_S4_L008_R1_001.fastq\n",
      "Approx 35% complete for TORN_Pool_4_S4_L008_R1_001.fastq\n",
      "Approx 40% complete for TORN_Pool_4_S4_L008_R1_001.fastq\n",
      "Approx 45% complete for TORN_Pool_4_S4_L008_R1_001.fastq\n",
      "Approx 50% complete for TORN_Pool_4_S4_L008_R1_001.fastq\n",
      "Approx 55% complete for TORN_Pool_4_S4_L008_R1_001.fastq\n",
      "Approx 60% complete for TORN_Pool_4_S4_L008_R1_001.fastq\n",
      "Approx 65% complete for TORN_Pool_4_S4_L008_R1_001.fastq\n",
      "Approx 70% complete for TORN_Pool_4_S4_L008_R1_001.fastq\n",
      "Approx 75% complete for TORN_Pool_4_S4_L008_R1_001.fastq\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Approx 80% complete for TORN_Pool_4_S4_L008_R1_001.fastq\n",
      "Approx 85% complete for TORN_Pool_4_S4_L008_R1_001.fastq\n",
      "Approx 90% complete for TORN_Pool_4_S4_L008_R1_001.fastq\n",
      "Approx 95% complete for TORN_Pool_4_S4_L008_R1_001.fastq\n",
      "Analysis complete for TORN_Pool_4_S4_L008_R1_001.fastq\n",
      "Started analysis of TORN_Pool_4_S4_L008_R2_001.fastq\n",
      "Approx 5% complete for TORN_Pool_4_S4_L008_R2_001.fastq\n",
      "Approx 10% complete for TORN_Pool_4_S4_L008_R2_001.fastq\n",
      "Approx 15% complete for TORN_Pool_4_S4_L008_R2_001.fastq\n",
      "Approx 20% complete for TORN_Pool_4_S4_L008_R2_001.fastq\n",
      "Approx 25% complete for TORN_Pool_4_S4_L008_R2_001.fastq\n",
      "Approx 30% complete for TORN_Pool_4_S4_L008_R2_001.fastq\n",
      "Approx 35% complete for TORN_Pool_4_S4_L008_R2_001.fastq\n",
      "Approx 40% complete for TORN_Pool_4_S4_L008_R2_001.fastq\n",
      "Approx 45% complete for TORN_Pool_4_S4_L008_R2_001.fastq\n",
      "Approx 50% complete for TORN_Pool_4_S4_L008_R2_001.fastq\n",
      "Approx 55% complete for TORN_Pool_4_S4_L008_R2_001.fastq\n",
      "Approx 60% complete for TORN_Pool_4_S4_L008_R2_001.fastq\n",
      "Approx 65% complete for TORN_Pool_4_S4_L008_R2_001.fastq\n",
      "Approx 70% complete for TORN_Pool_4_S4_L008_R2_001.fastq\n",
      "Approx 75% complete for TORN_Pool_4_S4_L008_R2_001.fastq\n",
      "Approx 80% complete for TORN_Pool_4_S4_L008_R2_001.fastq\n",
      "Approx 85% complete for TORN_Pool_4_S4_L008_R2_001.fastq\n",
      "Approx 90% complete for TORN_Pool_4_S4_L008_R2_001.fastq\n",
      "Approx 95% complete for TORN_Pool_4_S4_L008_R2_001.fastq\n",
      "Analysis complete for TORN_Pool_4_S4_L008_R2_001.fastq\n",
      "Started analysis of TORN_Pool_5_S5_L006_R1_001.fastq\n",
      "Approx 5% complete for TORN_Pool_5_S5_L006_R1_001.fastq\n",
      "Approx 10% complete for TORN_Pool_5_S5_L006_R1_001.fastq\n",
      "Approx 15% complete for TORN_Pool_5_S5_L006_R1_001.fastq\n",
      "Approx 20% complete for TORN_Pool_5_S5_L006_R1_001.fastq\n",
      "Approx 25% complete for TORN_Pool_5_S5_L006_R1_001.fastq\n",
      "Approx 30% complete for TORN_Pool_5_S5_L006_R1_001.fastq\n",
      "Approx 35% complete for TORN_Pool_5_S5_L006_R1_001.fastq\n",
      "Approx 40% complete for TORN_Pool_5_S5_L006_R1_001.fastq\n",
      "Approx 45% complete for TORN_Pool_5_S5_L006_R1_001.fastq\n",
      "Approx 50% complete for TORN_Pool_5_S5_L006_R1_001.fastq\n",
      "Approx 55% complete for TORN_Pool_5_S5_L006_R1_001.fastq\n",
      "Approx 60% complete for TORN_Pool_5_S5_L006_R1_001.fastq\n",
      "Approx 65% complete for TORN_Pool_5_S5_L006_R1_001.fastq\n",
      "Approx 70% complete for TORN_Pool_5_S5_L006_R1_001.fastq\n",
      "Approx 75% complete for TORN_Pool_5_S5_L006_R1_001.fastq\n",
      "Approx 80% complete for TORN_Pool_5_S5_L006_R1_001.fastq\n",
      "Approx 85% complete for TORN_Pool_5_S5_L006_R1_001.fastq\n",
      "Approx 90% complete for TORN_Pool_5_S5_L006_R1_001.fastq\n",
      "Approx 95% complete for TORN_Pool_5_S5_L006_R1_001.fastq\n",
      "Analysis complete for TORN_Pool_5_S5_L006_R1_001.fastq\n",
      "Started analysis of TORN_Pool_5_S5_L006_R2_001.fastq\n",
      "Approx 5% complete for TORN_Pool_5_S5_L006_R2_001.fastq\n",
      "Approx 10% complete for TORN_Pool_5_S5_L006_R2_001.fastq\n",
      "Approx 15% complete for TORN_Pool_5_S5_L006_R2_001.fastq\n",
      "Approx 20% complete for TORN_Pool_5_S5_L006_R2_001.fastq\n",
      "Approx 25% complete for TORN_Pool_5_S5_L006_R2_001.fastq\n",
      "Approx 30% complete for TORN_Pool_5_S5_L006_R2_001.fastq\n",
      "Approx 35% complete for TORN_Pool_5_S5_L006_R2_001.fastq\n",
      "Approx 40% complete for TORN_Pool_5_S5_L006_R2_001.fastq\n",
      "Approx 45% complete for TORN_Pool_5_S5_L006_R2_001.fastq\n",
      "Approx 50% complete for TORN_Pool_5_S5_L006_R2_001.fastq\n",
      "Approx 55% complete for TORN_Pool_5_S5_L006_R2_001.fastq\n",
      "Approx 60% complete for TORN_Pool_5_S5_L006_R2_001.fastq\n",
      "Approx 65% complete for TORN_Pool_5_S5_L006_R2_001.fastq\n",
      "Approx 70% complete for TORN_Pool_5_S5_L006_R2_001.fastq\n",
      "Approx 75% complete for TORN_Pool_5_S5_L006_R2_001.fastq\n",
      "Approx 80% complete for TORN_Pool_5_S5_L006_R2_001.fastq\n",
      "Approx 85% complete for TORN_Pool_5_S5_L006_R2_001.fastq\n",
      "Approx 90% complete for TORN_Pool_5_S5_L006_R2_001.fastq\n",
      "Approx 95% complete for TORN_Pool_5_S5_L006_R2_001.fastq\n",
      "Analysis complete for TORN_Pool_5_S5_L006_R2_001.fastq\n",
      "Started analysis of TORN_Pool_5_S5_L008_R1_001.fastq\n",
      "Approx 5% complete for TORN_Pool_5_S5_L008_R1_001.fastq\n",
      "Approx 10% complete for TORN_Pool_5_S5_L008_R1_001.fastq\n",
      "Approx 15% complete for TORN_Pool_5_S5_L008_R1_001.fastq\n",
      "Approx 20% complete for TORN_Pool_5_S5_L008_R1_001.fastq\n",
      "Approx 25% complete for TORN_Pool_5_S5_L008_R1_001.fastq\n",
      "Approx 30% complete for TORN_Pool_5_S5_L008_R1_001.fastq\n",
      "Approx 35% complete for TORN_Pool_5_S5_L008_R1_001.fastq\n",
      "Approx 40% complete for TORN_Pool_5_S5_L008_R1_001.fastq\n",
      "Approx 45% complete for TORN_Pool_5_S5_L008_R1_001.fastq\n",
      "Approx 50% complete for TORN_Pool_5_S5_L008_R1_001.fastq\n",
      "Approx 55% complete for TORN_Pool_5_S5_L008_R1_001.fastq\n",
      "Approx 60% complete for TORN_Pool_5_S5_L008_R1_001.fastq\n",
      "Approx 65% complete for TORN_Pool_5_S5_L008_R1_001.fastq\n",
      "Approx 70% complete for TORN_Pool_5_S5_L008_R1_001.fastq\n",
      "Approx 75% complete for TORN_Pool_5_S5_L008_R1_001.fastq\n",
      "Approx 80% complete for TORN_Pool_5_S5_L008_R1_001.fastq\n",
      "Approx 85% complete for TORN_Pool_5_S5_L008_R1_001.fastq\n",
      "Approx 90% complete for TORN_Pool_5_S5_L008_R1_001.fastq\n",
      "Approx 95% complete for TORN_Pool_5_S5_L008_R1_001.fastq\n",
      "Analysis complete for TORN_Pool_5_S5_L008_R1_001.fastq\n",
      "Started analysis of TORN_Pool_5_S5_L008_R2_001.fastq\n",
      "Approx 5% complete for TORN_Pool_5_S5_L008_R2_001.fastq\n",
      "Approx 10% complete for TORN_Pool_5_S5_L008_R2_001.fastq\n",
      "Approx 15% complete for TORN_Pool_5_S5_L008_R2_001.fastq\n",
      "Approx 20% complete for TORN_Pool_5_S5_L008_R2_001.fastq\n",
      "Approx 25% complete for TORN_Pool_5_S5_L008_R2_001.fastq\n",
      "Approx 30% complete for TORN_Pool_5_S5_L008_R2_001.fastq\n",
      "Approx 35% complete for TORN_Pool_5_S5_L008_R2_001.fastq\n",
      "Approx 40% complete for TORN_Pool_5_S5_L008_R2_001.fastq\n",
      "Approx 45% complete for TORN_Pool_5_S5_L008_R2_001.fastq\n",
      "Approx 50% complete for TORN_Pool_5_S5_L008_R2_001.fastq\n",
      "Approx 55% complete for TORN_Pool_5_S5_L008_R2_001.fastq\n",
      "Approx 60% complete for TORN_Pool_5_S5_L008_R2_001.fastq\n",
      "Approx 65% complete for TORN_Pool_5_S5_L008_R2_001.fastq\n",
      "Approx 70% complete for TORN_Pool_5_S5_L008_R2_001.fastq\n",
      "Approx 75% complete for TORN_Pool_5_S5_L008_R2_001.fastq\n",
      "Approx 80% complete for TORN_Pool_5_S5_L008_R2_001.fastq\n",
      "Approx 85% complete for TORN_Pool_5_S5_L008_R2_001.fastq\n",
      "Approx 90% complete for TORN_Pool_5_S5_L008_R2_001.fastq\n",
      "Approx 95% complete for TORN_Pool_5_S5_L008_R2_001.fastq\n",
      "Analysis complete for TORN_Pool_5_S5_L008_R2_001.fastq\n",
      "Started analysis of TORN_Pool_6_S6_L006_R1_001.fastq\n",
      "Approx 5% complete for TORN_Pool_6_S6_L006_R1_001.fastq\n",
      "Approx 10% complete for TORN_Pool_6_S6_L006_R1_001.fastq\n",
      "Approx 15% complete for TORN_Pool_6_S6_L006_R1_001.fastq\n",
      "Approx 20% complete for TORN_Pool_6_S6_L006_R1_001.fastq\n",
      "Approx 25% complete for TORN_Pool_6_S6_L006_R1_001.fastq\n",
      "Approx 30% complete for TORN_Pool_6_S6_L006_R1_001.fastq\n",
      "Approx 35% complete for TORN_Pool_6_S6_L006_R1_001.fastq\n",
      "Approx 40% complete for TORN_Pool_6_S6_L006_R1_001.fastq\n",
      "Approx 45% complete for TORN_Pool_6_S6_L006_R1_001.fastq\n",
      "Approx 50% complete for TORN_Pool_6_S6_L006_R1_001.fastq\n",
      "Approx 55% complete for TORN_Pool_6_S6_L006_R1_001.fastq\n",
      "Approx 60% complete for TORN_Pool_6_S6_L006_R1_001.fastq\n",
      "Approx 65% complete for TORN_Pool_6_S6_L006_R1_001.fastq\n",
      "Approx 70% complete for TORN_Pool_6_S6_L006_R1_001.fastq\n",
      "Approx 75% complete for TORN_Pool_6_S6_L006_R1_001.fastq\n",
      "Approx 80% complete for TORN_Pool_6_S6_L006_R1_001.fastq\n",
      "Approx 85% complete for TORN_Pool_6_S6_L006_R1_001.fastq\n",
      "Approx 90% complete for TORN_Pool_6_S6_L006_R1_001.fastq\n",
      "Approx 95% complete for TORN_Pool_6_S6_L006_R1_001.fastq\n",
      "Analysis complete for TORN_Pool_6_S6_L006_R1_001.fastq\n",
      "Started analysis of TORN_Pool_6_S6_L006_R2_001.fastq\n",
      "Approx 5% complete for TORN_Pool_6_S6_L006_R2_001.fastq\n",
      "Approx 10% complete for TORN_Pool_6_S6_L006_R2_001.fastq\n",
      "Approx 15% complete for TORN_Pool_6_S6_L006_R2_001.fastq\n",
      "Approx 20% complete for TORN_Pool_6_S6_L006_R2_001.fastq\n",
      "Approx 25% complete for TORN_Pool_6_S6_L006_R2_001.fastq\n",
      "Approx 30% complete for TORN_Pool_6_S6_L006_R2_001.fastq\n",
      "Approx 35% complete for TORN_Pool_6_S6_L006_R2_001.fastq\n",
      "Approx 40% complete for TORN_Pool_6_S6_L006_R2_001.fastq\n",
      "Approx 45% complete for TORN_Pool_6_S6_L006_R2_001.fastq\n",
      "Approx 50% complete for TORN_Pool_6_S6_L006_R2_001.fastq\n",
      "Approx 55% complete for TORN_Pool_6_S6_L006_R2_001.fastq\n",
      "Approx 60% complete for TORN_Pool_6_S6_L006_R2_001.fastq\n",
      "Approx 65% complete for TORN_Pool_6_S6_L006_R2_001.fastq\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Approx 70% complete for TORN_Pool_6_S6_L006_R2_001.fastq\n",
      "Approx 75% complete for TORN_Pool_6_S6_L006_R2_001.fastq\n",
      "Approx 80% complete for TORN_Pool_6_S6_L006_R2_001.fastq\n",
      "Approx 85% complete for TORN_Pool_6_S6_L006_R2_001.fastq\n",
      "Approx 90% complete for TORN_Pool_6_S6_L006_R2_001.fastq\n",
      "Approx 95% complete for TORN_Pool_6_S6_L006_R2_001.fastq\n",
      "Analysis complete for TORN_Pool_6_S6_L006_R2_001.fastq\n",
      "Started analysis of TORN_Pool_6_S6_L008_R1_001.fastq\n",
      "Approx 5% complete for TORN_Pool_6_S6_L008_R1_001.fastq\n",
      "Approx 10% complete for TORN_Pool_6_S6_L008_R1_001.fastq\n",
      "Approx 15% complete for TORN_Pool_6_S6_L008_R1_001.fastq\n",
      "Approx 20% complete for TORN_Pool_6_S6_L008_R1_001.fastq\n",
      "Approx 25% complete for TORN_Pool_6_S6_L008_R1_001.fastq\n",
      "Approx 30% complete for TORN_Pool_6_S6_L008_R1_001.fastq\n",
      "Approx 35% complete for TORN_Pool_6_S6_L008_R1_001.fastq\n",
      "Approx 40% complete for TORN_Pool_6_S6_L008_R1_001.fastq\n",
      "Approx 45% complete for TORN_Pool_6_S6_L008_R1_001.fastq\n",
      "Approx 50% complete for TORN_Pool_6_S6_L008_R1_001.fastq\n",
      "Approx 55% complete for TORN_Pool_6_S6_L008_R1_001.fastq\n",
      "Approx 60% complete for TORN_Pool_6_S6_L008_R1_001.fastq\n",
      "Approx 65% complete for TORN_Pool_6_S6_L008_R1_001.fastq\n",
      "Approx 70% complete for TORN_Pool_6_S6_L008_R1_001.fastq\n",
      "Approx 75% complete for TORN_Pool_6_S6_L008_R1_001.fastq\n",
      "Approx 80% complete for TORN_Pool_6_S6_L008_R1_001.fastq\n",
      "Approx 85% complete for TORN_Pool_6_S6_L008_R1_001.fastq\n",
      "Approx 90% complete for TORN_Pool_6_S6_L008_R1_001.fastq\n",
      "Approx 95% complete for TORN_Pool_6_S6_L008_R1_001.fastq\n",
      "Analysis complete for TORN_Pool_6_S6_L008_R1_001.fastq\n",
      "Started analysis of TORN_Pool_6_S6_L008_R2_001.fastq\n",
      "Approx 5% complete for TORN_Pool_6_S6_L008_R2_001.fastq\n",
      "Approx 10% complete for TORN_Pool_6_S6_L008_R2_001.fastq\n",
      "Approx 15% complete for TORN_Pool_6_S6_L008_R2_001.fastq\n",
      "Approx 20% complete for TORN_Pool_6_S6_L008_R2_001.fastq\n",
      "Approx 25% complete for TORN_Pool_6_S6_L008_R2_001.fastq\n",
      "Approx 30% complete for TORN_Pool_6_S6_L008_R2_001.fastq\n",
      "Approx 35% complete for TORN_Pool_6_S6_L008_R2_001.fastq\n",
      "Approx 40% complete for TORN_Pool_6_S6_L008_R2_001.fastq\n",
      "Approx 45% complete for TORN_Pool_6_S6_L008_R2_001.fastq\n",
      "Approx 50% complete for TORN_Pool_6_S6_L008_R2_001.fastq\n",
      "Approx 55% complete for TORN_Pool_6_S6_L008_R2_001.fastq\n",
      "Approx 60% complete for TORN_Pool_6_S6_L008_R2_001.fastq\n",
      "Approx 65% complete for TORN_Pool_6_S6_L008_R2_001.fastq\n",
      "Approx 70% complete for TORN_Pool_6_S6_L008_R2_001.fastq\n",
      "Approx 75% complete for TORN_Pool_6_S6_L008_R2_001.fastq\n",
      "Approx 80% complete for TORN_Pool_6_S6_L008_R2_001.fastq\n",
      "Approx 85% complete for TORN_Pool_6_S6_L008_R2_001.fastq\n",
      "Approx 90% complete for TORN_Pool_6_S6_L008_R2_001.fastq\n",
      "Approx 95% complete for TORN_Pool_6_S6_L008_R2_001.fastq\n",
      "Analysis complete for TORN_Pool_6_S6_L008_R2_001.fastq\n",
      "Started analysis of TORN_Pool_7_S7_L006_R1_001.fastq\n",
      "Approx 5% complete for TORN_Pool_7_S7_L006_R1_001.fastq\n",
      "Approx 10% complete for TORN_Pool_7_S7_L006_R1_001.fastq\n",
      "Approx 15% complete for TORN_Pool_7_S7_L006_R1_001.fastq\n",
      "Approx 20% complete for TORN_Pool_7_S7_L006_R1_001.fastq\n",
      "Approx 25% complete for TORN_Pool_7_S7_L006_R1_001.fastq\n",
      "Approx 30% complete for TORN_Pool_7_S7_L006_R1_001.fastq\n",
      "Approx 35% complete for TORN_Pool_7_S7_L006_R1_001.fastq\n",
      "Approx 40% complete for TORN_Pool_7_S7_L006_R1_001.fastq\n",
      "Approx 45% complete for TORN_Pool_7_S7_L006_R1_001.fastq\n",
      "Approx 50% complete for TORN_Pool_7_S7_L006_R1_001.fastq\n",
      "Approx 55% complete for TORN_Pool_7_S7_L006_R1_001.fastq\n",
      "Approx 60% complete for TORN_Pool_7_S7_L006_R1_001.fastq\n",
      "Approx 65% complete for TORN_Pool_7_S7_L006_R1_001.fastq\n",
      "Approx 70% complete for TORN_Pool_7_S7_L006_R1_001.fastq\n",
      "Approx 75% complete for TORN_Pool_7_S7_L006_R1_001.fastq\n",
      "Approx 80% complete for TORN_Pool_7_S7_L006_R1_001.fastq\n",
      "Approx 85% complete for TORN_Pool_7_S7_L006_R1_001.fastq\n",
      "Approx 90% complete for TORN_Pool_7_S7_L006_R1_001.fastq\n",
      "Approx 95% complete for TORN_Pool_7_S7_L006_R1_001.fastq\n",
      "Analysis complete for TORN_Pool_7_S7_L006_R1_001.fastq\n",
      "Started analysis of TORN_Pool_7_S7_L006_R2_001.fastq\n",
      "Approx 5% complete for TORN_Pool_7_S7_L006_R2_001.fastq\n",
      "Approx 10% complete for TORN_Pool_7_S7_L006_R2_001.fastq\n",
      "Approx 15% complete for TORN_Pool_7_S7_L006_R2_001.fastq\n",
      "Approx 20% complete for TORN_Pool_7_S7_L006_R2_001.fastq\n",
      "Approx 25% complete for TORN_Pool_7_S7_L006_R2_001.fastq\n",
      "Approx 30% complete for TORN_Pool_7_S7_L006_R2_001.fastq\n",
      "Approx 35% complete for TORN_Pool_7_S7_L006_R2_001.fastq\n",
      "Approx 40% complete for TORN_Pool_7_S7_L006_R2_001.fastq\n",
      "Approx 45% complete for TORN_Pool_7_S7_L006_R2_001.fastq\n",
      "Approx 50% complete for TORN_Pool_7_S7_L006_R2_001.fastq\n",
      "Approx 55% complete for TORN_Pool_7_S7_L006_R2_001.fastq\n",
      "Approx 60% complete for TORN_Pool_7_S7_L006_R2_001.fastq\n",
      "Approx 65% complete for TORN_Pool_7_S7_L006_R2_001.fastq\n",
      "Approx 70% complete for TORN_Pool_7_S7_L006_R2_001.fastq\n",
      "Approx 75% complete for TORN_Pool_7_S7_L006_R2_001.fastq\n",
      "Approx 80% complete for TORN_Pool_7_S7_L006_R2_001.fastq\n",
      "Approx 85% complete for TORN_Pool_7_S7_L006_R2_001.fastq\n",
      "Approx 90% complete for TORN_Pool_7_S7_L006_R2_001.fastq\n",
      "Approx 95% complete for TORN_Pool_7_S7_L006_R2_001.fastq\n",
      "Analysis complete for TORN_Pool_7_S7_L006_R2_001.fastq\n",
      "Started analysis of TORN_Pool_7_S7_L008_R1_001.fastq\n",
      "Approx 5% complete for TORN_Pool_7_S7_L008_R1_001.fastq\n",
      "Approx 10% complete for TORN_Pool_7_S7_L008_R1_001.fastq\n",
      "Approx 15% complete for TORN_Pool_7_S7_L008_R1_001.fastq\n",
      "Approx 20% complete for TORN_Pool_7_S7_L008_R1_001.fastq\n",
      "Approx 25% complete for TORN_Pool_7_S7_L008_R1_001.fastq\n",
      "Approx 30% complete for TORN_Pool_7_S7_L008_R1_001.fastq\n",
      "Approx 35% complete for TORN_Pool_7_S7_L008_R1_001.fastq\n",
      "Approx 40% complete for TORN_Pool_7_S7_L008_R1_001.fastq\n",
      "Approx 45% complete for TORN_Pool_7_S7_L008_R1_001.fastq\n",
      "Approx 50% complete for TORN_Pool_7_S7_L008_R1_001.fastq\n",
      "Approx 55% complete for TORN_Pool_7_S7_L008_R1_001.fastq\n",
      "Approx 60% complete for TORN_Pool_7_S7_L008_R1_001.fastq\n",
      "Approx 65% complete for TORN_Pool_7_S7_L008_R1_001.fastq\n",
      "Approx 70% complete for TORN_Pool_7_S7_L008_R1_001.fastq\n",
      "Approx 75% complete for TORN_Pool_7_S7_L008_R1_001.fastq\n",
      "Approx 80% complete for TORN_Pool_7_S7_L008_R1_001.fastq\n",
      "Approx 85% complete for TORN_Pool_7_S7_L008_R1_001.fastq\n",
      "Approx 90% complete for TORN_Pool_7_S7_L008_R1_001.fastq\n",
      "Approx 95% complete for TORN_Pool_7_S7_L008_R1_001.fastq\n",
      "Analysis complete for TORN_Pool_7_S7_L008_R1_001.fastq\n",
      "Started analysis of TORN_Pool_7_S7_L008_R2_001.fastq\n",
      "Approx 5% complete for TORN_Pool_7_S7_L008_R2_001.fastq\n",
      "Approx 10% complete for TORN_Pool_7_S7_L008_R2_001.fastq\n",
      "Approx 15% complete for TORN_Pool_7_S7_L008_R2_001.fastq\n",
      "Approx 20% complete for TORN_Pool_7_S7_L008_R2_001.fastq\n",
      "Approx 25% complete for TORN_Pool_7_S7_L008_R2_001.fastq\n",
      "Approx 30% complete for TORN_Pool_7_S7_L008_R2_001.fastq\n",
      "Approx 35% complete for TORN_Pool_7_S7_L008_R2_001.fastq\n",
      "Approx 40% complete for TORN_Pool_7_S7_L008_R2_001.fastq\n",
      "Approx 45% complete for TORN_Pool_7_S7_L008_R2_001.fastq\n",
      "Approx 50% complete for TORN_Pool_7_S7_L008_R2_001.fastq\n",
      "Approx 55% complete for TORN_Pool_7_S7_L008_R2_001.fastq\n",
      "Approx 60% complete for TORN_Pool_7_S7_L008_R2_001.fastq\n",
      "Approx 65% complete for TORN_Pool_7_S7_L008_R2_001.fastq\n",
      "Approx 70% complete for TORN_Pool_7_S7_L008_R2_001.fastq\n",
      "Approx 75% complete for TORN_Pool_7_S7_L008_R2_001.fastq\n",
      "Approx 80% complete for TORN_Pool_7_S7_L008_R2_001.fastq\n",
      "Approx 85% complete for TORN_Pool_7_S7_L008_R2_001.fastq\n",
      "Approx 90% complete for TORN_Pool_7_S7_L008_R2_001.fastq\n",
      "Approx 95% complete for TORN_Pool_7_S7_L008_R2_001.fastq\n",
      "Analysis complete for TORN_Pool_7_S7_L008_R2_001.fastq\n",
      "Started analysis of TORN_Pool_8_S8_L006_R1_001.fastq\n",
      "Approx 5% complete for TORN_Pool_8_S8_L006_R1_001.fastq\n",
      "Approx 10% complete for TORN_Pool_8_S8_L006_R1_001.fastq\n",
      "Approx 15% complete for TORN_Pool_8_S8_L006_R1_001.fastq\n",
      "Approx 20% complete for TORN_Pool_8_S8_L006_R1_001.fastq\n",
      "Approx 25% complete for TORN_Pool_8_S8_L006_R1_001.fastq\n",
      "Approx 30% complete for TORN_Pool_8_S8_L006_R1_001.fastq\n",
      "Approx 35% complete for TORN_Pool_8_S8_L006_R1_001.fastq\n",
      "Approx 40% complete for TORN_Pool_8_S8_L006_R1_001.fastq\n",
      "Approx 45% complete for TORN_Pool_8_S8_L006_R1_001.fastq\n",
      "Approx 50% complete for TORN_Pool_8_S8_L006_R1_001.fastq\n",
      "Approx 55% complete for TORN_Pool_8_S8_L006_R1_001.fastq\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Approx 60% complete for TORN_Pool_8_S8_L006_R1_001.fastq\n",
      "Approx 65% complete for TORN_Pool_8_S8_L006_R1_001.fastq\n",
      "Approx 70% complete for TORN_Pool_8_S8_L006_R1_001.fastq\n",
      "Approx 75% complete for TORN_Pool_8_S8_L006_R1_001.fastq\n",
      "Approx 80% complete for TORN_Pool_8_S8_L006_R1_001.fastq\n",
      "Approx 85% complete for TORN_Pool_8_S8_L006_R1_001.fastq\n",
      "Approx 90% complete for TORN_Pool_8_S8_L006_R1_001.fastq\n",
      "Approx 95% complete for TORN_Pool_8_S8_L006_R1_001.fastq\n",
      "Analysis complete for TORN_Pool_8_S8_L006_R1_001.fastq\n",
      "Started analysis of TORN_Pool_8_S8_L006_R2_001.fastq\n",
      "Approx 5% complete for TORN_Pool_8_S8_L006_R2_001.fastq\n",
      "Approx 10% complete for TORN_Pool_8_S8_L006_R2_001.fastq\n",
      "Approx 15% complete for TORN_Pool_8_S8_L006_R2_001.fastq\n",
      "Approx 20% complete for TORN_Pool_8_S8_L006_R2_001.fastq\n",
      "Approx 25% complete for TORN_Pool_8_S8_L006_R2_001.fastq\n",
      "Approx 30% complete for TORN_Pool_8_S8_L006_R2_001.fastq\n",
      "Approx 35% complete for TORN_Pool_8_S8_L006_R2_001.fastq\n",
      "Approx 40% complete for TORN_Pool_8_S8_L006_R2_001.fastq\n",
      "Approx 45% complete for TORN_Pool_8_S8_L006_R2_001.fastq\n",
      "Approx 50% complete for TORN_Pool_8_S8_L006_R2_001.fastq\n",
      "Approx 55% complete for TORN_Pool_8_S8_L006_R2_001.fastq\n",
      "Approx 60% complete for TORN_Pool_8_S8_L006_R2_001.fastq\n",
      "Approx 65% complete for TORN_Pool_8_S8_L006_R2_001.fastq\n",
      "Approx 70% complete for TORN_Pool_8_S8_L006_R2_001.fastq\n",
      "Approx 75% complete for TORN_Pool_8_S8_L006_R2_001.fastq\n",
      "Approx 80% complete for TORN_Pool_8_S8_L006_R2_001.fastq\n",
      "Approx 85% complete for TORN_Pool_8_S8_L006_R2_001.fastq\n",
      "Approx 90% complete for TORN_Pool_8_S8_L006_R2_001.fastq\n",
      "Approx 95% complete for TORN_Pool_8_S8_L006_R2_001.fastq\n",
      "Analysis complete for TORN_Pool_8_S8_L006_R2_001.fastq\n",
      "Started analysis of TORN_Pool_8_S8_L008_R1_001.fastq\n",
      "Approx 5% complete for TORN_Pool_8_S8_L008_R1_001.fastq\n",
      "Approx 10% complete for TORN_Pool_8_S8_L008_R1_001.fastq\n",
      "Approx 15% complete for TORN_Pool_8_S8_L008_R1_001.fastq\n",
      "Approx 20% complete for TORN_Pool_8_S8_L008_R1_001.fastq\n",
      "Approx 25% complete for TORN_Pool_8_S8_L008_R1_001.fastq\n",
      "Approx 30% complete for TORN_Pool_8_S8_L008_R1_001.fastq\n",
      "Approx 35% complete for TORN_Pool_8_S8_L008_R1_001.fastq\n",
      "Approx 40% complete for TORN_Pool_8_S8_L008_R1_001.fastq\n",
      "Approx 45% complete for TORN_Pool_8_S8_L008_R1_001.fastq\n",
      "Approx 50% complete for TORN_Pool_8_S8_L008_R1_001.fastq\n",
      "Approx 55% complete for TORN_Pool_8_S8_L008_R1_001.fastq\n",
      "Approx 60% complete for TORN_Pool_8_S8_L008_R1_001.fastq\n",
      "Approx 65% complete for TORN_Pool_8_S8_L008_R1_001.fastq\n",
      "Approx 70% complete for TORN_Pool_8_S8_L008_R1_001.fastq\n",
      "Approx 75% complete for TORN_Pool_8_S8_L008_R1_001.fastq\n",
      "Approx 80% complete for TORN_Pool_8_S8_L008_R1_001.fastq\n",
      "Approx 85% complete for TORN_Pool_8_S8_L008_R1_001.fastq\n",
      "Approx 90% complete for TORN_Pool_8_S8_L008_R1_001.fastq\n",
      "Approx 95% complete for TORN_Pool_8_S8_L008_R1_001.fastq\n",
      "Analysis complete for TORN_Pool_8_S8_L008_R1_001.fastq\n",
      "Started analysis of TORN_Pool_8_S8_L008_R2_001.fastq\n",
      "Approx 5% complete for TORN_Pool_8_S8_L008_R2_001.fastq\n",
      "Approx 10% complete for TORN_Pool_8_S8_L008_R2_001.fastq\n",
      "Approx 15% complete for TORN_Pool_8_S8_L008_R2_001.fastq\n",
      "Approx 20% complete for TORN_Pool_8_S8_L008_R2_001.fastq\n",
      "Approx 25% complete for TORN_Pool_8_S8_L008_R2_001.fastq\n",
      "Approx 30% complete for TORN_Pool_8_S8_L008_R2_001.fastq\n",
      "Approx 35% complete for TORN_Pool_8_S8_L008_R2_001.fastq\n",
      "Approx 40% complete for TORN_Pool_8_S8_L008_R2_001.fastq\n",
      "Approx 45% complete for TORN_Pool_8_S8_L008_R2_001.fastq\n",
      "Approx 50% complete for TORN_Pool_8_S8_L008_R2_001.fastq\n",
      "Approx 55% complete for TORN_Pool_8_S8_L008_R2_001.fastq\n",
      "Approx 60% complete for TORN_Pool_8_S8_L008_R2_001.fastq\n",
      "Approx 65% complete for TORN_Pool_8_S8_L008_R2_001.fastq\n",
      "Approx 70% complete for TORN_Pool_8_S8_L008_R2_001.fastq\n",
      "Approx 75% complete for TORN_Pool_8_S8_L008_R2_001.fastq\n",
      "Approx 80% complete for TORN_Pool_8_S8_L008_R2_001.fastq\n",
      "Approx 85% complete for TORN_Pool_8_S8_L008_R2_001.fastq\n",
      "Approx 90% complete for TORN_Pool_8_S8_L008_R2_001.fastq\n",
      "Approx 95% complete for TORN_Pool_8_S8_L008_R2_001.fastq\n",
      "Analysis complete for TORN_Pool_8_S8_L008_R2_001.fastq\n",
      "Started analysis of TORN_Pool_9_S9_L006_R1_001.fastq\n",
      "Approx 5% complete for TORN_Pool_9_S9_L006_R1_001.fastq\n",
      "Approx 10% complete for TORN_Pool_9_S9_L006_R1_001.fastq\n",
      "Approx 15% complete for TORN_Pool_9_S9_L006_R1_001.fastq\n",
      "Approx 20% complete for TORN_Pool_9_S9_L006_R1_001.fastq\n",
      "Approx 25% complete for TORN_Pool_9_S9_L006_R1_001.fastq\n",
      "Approx 30% complete for TORN_Pool_9_S9_L006_R1_001.fastq\n",
      "Approx 35% complete for TORN_Pool_9_S9_L006_R1_001.fastq\n",
      "Approx 40% complete for TORN_Pool_9_S9_L006_R1_001.fastq\n",
      "Approx 45% complete for TORN_Pool_9_S9_L006_R1_001.fastq\n",
      "Approx 50% complete for TORN_Pool_9_S9_L006_R1_001.fastq\n",
      "Approx 55% complete for TORN_Pool_9_S9_L006_R1_001.fastq\n",
      "Approx 60% complete for TORN_Pool_9_S9_L006_R1_001.fastq\n",
      "Approx 65% complete for TORN_Pool_9_S9_L006_R1_001.fastq\n",
      "Approx 70% complete for TORN_Pool_9_S9_L006_R1_001.fastq\n",
      "Approx 75% complete for TORN_Pool_9_S9_L006_R1_001.fastq\n",
      "Approx 80% complete for TORN_Pool_9_S9_L006_R1_001.fastq\n",
      "Approx 85% complete for TORN_Pool_9_S9_L006_R1_001.fastq\n",
      "Approx 90% complete for TORN_Pool_9_S9_L006_R1_001.fastq\n",
      "Approx 95% complete for TORN_Pool_9_S9_L006_R1_001.fastq\n",
      "Analysis complete for TORN_Pool_9_S9_L006_R1_001.fastq\n",
      "Started analysis of TORN_Pool_9_S9_L006_R2_001.fastq\n",
      "Approx 5% complete for TORN_Pool_9_S9_L006_R2_001.fastq\n",
      "Approx 10% complete for TORN_Pool_9_S9_L006_R2_001.fastq\n",
      "Approx 15% complete for TORN_Pool_9_S9_L006_R2_001.fastq\n",
      "Approx 20% complete for TORN_Pool_9_S9_L006_R2_001.fastq\n",
      "Approx 25% complete for TORN_Pool_9_S9_L006_R2_001.fastq\n",
      "Approx 30% complete for TORN_Pool_9_S9_L006_R2_001.fastq\n",
      "Approx 35% complete for TORN_Pool_9_S9_L006_R2_001.fastq\n",
      "Approx 40% complete for TORN_Pool_9_S9_L006_R2_001.fastq\n",
      "Approx 45% complete for TORN_Pool_9_S9_L006_R2_001.fastq\n",
      "Approx 50% complete for TORN_Pool_9_S9_L006_R2_001.fastq\n",
      "Approx 55% complete for TORN_Pool_9_S9_L006_R2_001.fastq\n",
      "Approx 60% complete for TORN_Pool_9_S9_L006_R2_001.fastq\n",
      "Approx 65% complete for TORN_Pool_9_S9_L006_R2_001.fastq\n",
      "Approx 70% complete for TORN_Pool_9_S9_L006_R2_001.fastq\n",
      "Approx 75% complete for TORN_Pool_9_S9_L006_R2_001.fastq\n",
      "Approx 80% complete for TORN_Pool_9_S9_L006_R2_001.fastq\n",
      "Approx 85% complete for TORN_Pool_9_S9_L006_R2_001.fastq\n",
      "Approx 90% complete for TORN_Pool_9_S9_L006_R2_001.fastq\n",
      "Approx 95% complete for TORN_Pool_9_S9_L006_R2_001.fastq\n",
      "Analysis complete for TORN_Pool_9_S9_L006_R2_001.fastq\n",
      "Started analysis of TORN_Pool_9_S9_L008_R1_001.fastq\n",
      "Approx 5% complete for TORN_Pool_9_S9_L008_R1_001.fastq\n",
      "Approx 10% complete for TORN_Pool_9_S9_L008_R1_001.fastq\n",
      "Approx 15% complete for TORN_Pool_9_S9_L008_R1_001.fastq\n",
      "Approx 20% complete for TORN_Pool_9_S9_L008_R1_001.fastq\n",
      "Approx 25% complete for TORN_Pool_9_S9_L008_R1_001.fastq\n",
      "Approx 30% complete for TORN_Pool_9_S9_L008_R1_001.fastq\n",
      "Approx 35% complete for TORN_Pool_9_S9_L008_R1_001.fastq\n",
      "Approx 40% complete for TORN_Pool_9_S9_L008_R1_001.fastq\n",
      "Approx 45% complete for TORN_Pool_9_S9_L008_R1_001.fastq\n",
      "Approx 50% complete for TORN_Pool_9_S9_L008_R1_001.fastq\n",
      "Approx 55% complete for TORN_Pool_9_S9_L008_R1_001.fastq\n",
      "Approx 60% complete for TORN_Pool_9_S9_L008_R1_001.fastq\n",
      "Approx 65% complete for TORN_Pool_9_S9_L008_R1_001.fastq\n",
      "Approx 70% complete for TORN_Pool_9_S9_L008_R1_001.fastq\n",
      "Approx 75% complete for TORN_Pool_9_S9_L008_R1_001.fastq\n",
      "Approx 80% complete for TORN_Pool_9_S9_L008_R1_001.fastq\n",
      "Approx 85% complete for TORN_Pool_9_S9_L008_R1_001.fastq\n",
      "Approx 90% complete for TORN_Pool_9_S9_L008_R1_001.fastq\n",
      "Approx 95% complete for TORN_Pool_9_S9_L008_R1_001.fastq\n",
      "Analysis complete for TORN_Pool_9_S9_L008_R1_001.fastq\n",
      "Started analysis of TORN_Pool_9_S9_L008_R2_001.fastq\n",
      "Approx 5% complete for TORN_Pool_9_S9_L008_R2_001.fastq\n",
      "Approx 10% complete for TORN_Pool_9_S9_L008_R2_001.fastq\n",
      "Approx 15% complete for TORN_Pool_9_S9_L008_R2_001.fastq\n",
      "Approx 20% complete for TORN_Pool_9_S9_L008_R2_001.fastq\n",
      "Approx 25% complete for TORN_Pool_9_S9_L008_R2_001.fastq\n",
      "Approx 30% complete for TORN_Pool_9_S9_L008_R2_001.fastq\n",
      "Approx 35% complete for TORN_Pool_9_S9_L008_R2_001.fastq\n",
      "Approx 40% complete for TORN_Pool_9_S9_L008_R2_001.fastq\n",
      "Approx 45% complete for TORN_Pool_9_S9_L008_R2_001.fastq\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Approx 50% complete for TORN_Pool_9_S9_L008_R2_001.fastq\n",
      "Approx 55% complete for TORN_Pool_9_S9_L008_R2_001.fastq\n",
      "Approx 60% complete for TORN_Pool_9_S9_L008_R2_001.fastq\n",
      "Approx 65% complete for TORN_Pool_9_S9_L008_R2_001.fastq\n",
      "Approx 70% complete for TORN_Pool_9_S9_L008_R2_001.fastq\n",
      "Approx 75% complete for TORN_Pool_9_S9_L008_R2_001.fastq\n",
      "Approx 80% complete for TORN_Pool_9_S9_L008_R2_001.fastq\n",
      "Approx 85% complete for TORN_Pool_9_S9_L008_R2_001.fastq\n",
      "Approx 90% complete for TORN_Pool_9_S9_L008_R2_001.fastq\n",
      "Approx 95% complete for TORN_Pool_9_S9_L008_R2_001.fastq\n",
      "Analysis complete for TORN_Pool_9_S9_L008_R2_001.fastq\n"
     ]
    }
   ],
   "source": [
    "!{biotools}FastQC/fastqc \\\n",
    "{project}data/*.fastq \\\n",
    "-o {project}analysis/fastqc"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Produces an html file which you can open in a browser."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Merge files from multiple lanes\n",
    "Function: Merge the data on lane6 and lane8 together  \n",
    "Input file: unziped fastq files(.fastq)  \n",
    "Output file: merged files(.fastq)\n",
    "\n",
    "\n",
    "Let's do a test run on one set of files."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [],
   "source": [
    "!cat \\\n",
    "{project}data/TORN_Pool_4_S4_L006_R1_001.fastq \\\n",
    "{project}data/TORN_Pool_4_S4_L008_R1_001.fastq \\\n",
    "> {project}analysis/merge/TORN_Pool_4_S4_Merged_R1_001.fastq"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now let's try to do the rest using a loop (One for R1 and one of R2). For this I navigated to the data/ folder in Terminal and used the following lines of code. I couldn't figure out how to use absolute paths in Jupyter. The output files went into data/ but I moved them into analysis/merge/. New output file name convention is TORN_Pool_#_S#_R#.fastq."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "!(for i in *_L006_R1_001.fastq; do cat ${i%_L006_R1_001.fastq}_L006_R1_001.fastq ${i%_L006_R1_001.fastq}_L008_R1_001.fastq > ${i%_L006_R1_001.fastq}_R1.fastq; done)\n",
    "!(for i in *_L006_R2_001.fastq; do cat ${i%_L006_R2_001.fastq}_L006_R2_001.fastq ${i%_L006_R2_001.fastq}_L008_R2_001.fastq > ${i%_L006_R2_001.fastq}_R2.fastq; done)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Trim-Galore"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Need to verify cutadapt and trimgalore are up-to-date."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1.18\r\n"
     ]
    }
   ],
   "source": [
    "!{biotools}cutadapt-master/cutadapt --version"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\r\n",
      "                        Quality-/Adapter-/RRBS-/Hard-Trimming\r\n",
      "                                (powered by Cutadapt)\r\n",
      "                                  version 0.5.0\r\n",
      "\r\n",
      "                               Last update: 28 06 2018\r\n",
      "\r\n"
     ]
    }
   ],
   "source": [
    "!{biotools}TrimGalore-0.5.0/trim_galore --version"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Tried to run TrimGalore on one sample, but couldn't get to work. Need to try the for loop in source material."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "Path to Cutadapt set as: '/Applications/bio-tools/cutadapt-master/cutadapt' (user defined)\n",
      "1.18\n",
      "Cutadapt seems to be working fine (tested command '/Applications/bio-tools/cutadapt-master/cutadapt --version')\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> /Users/calderatta/Desktop/FISH 546 - Bioinformatics/project/data/TORN_Pool_10_S10_L008_R2_001.fastq <<)\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "Illumina\t62856\tAGATCGGAAGAGC\t1000000\t6.29\n",
      "Nextera\t6\tCTGTCTCTTATA\t1000000\t0.00\n",
      "smallRNA\t1\tTGGAATTCTCGG\t1000000\t0.00\n",
      "Using Illumina adapter for trimming (count: 62856). Second best hit was Nextera (count: 6)\n",
      "\n",
      "Writing report to '/Users/calderatta/Desktop/FISH 546 - Bioinformatics/project/analysis/trimgalore/TORN_Pool_10_S10_L008_R2_001.fastq_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: /Users/calderatta/Desktop/FISH 546 - Bioinformatics/project/data/TORN_Pool_10_S10_L008_R2_001.fastq\n",
      "Trimming mode: single-end\n",
      "Trim Galore version: 0.5.0\n",
      "Cutadapt version: 1.18\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length before a sequence gets removed: 20 bp\n",
      "\n",
      "Writing final adapter and quality trimmed output to TORN_Pool_10_S10_L008_R2_001_trimmed.fq\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff 20) and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file /Users/calderatta/Desktop/FISH 546 - Bioinformatics/project/data/TORN_Pool_10_S10_L008_R2_001.fastq <<< \n",
      "Run \"cutadapt --help\" to see command-line options.\n",
      "See https://cutadapt.readthedocs.io/ for full documentation.\n",
      "\n",
      "cutadapt: error: Too many parameters.\n",
      "\n",
      "\n",
      "Cutadapt terminated with exit signal: '512'.\n",
      "Terminating Trim Galore run, please check error message(s) to get an idea what went wrong...\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!{biotools}TrimGalore-0.5.0/trim_galore \\\n",
    "{project}data/TORN_Pool_10_S10_L008_R2_001.fastq \\\n",
    "--path_to_cutadapt {biotools}cutadapt-master/cutadapt \\\n",
    "-o {project}analysis/trimgalore/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Function: Trim the adapter and low quality reads in fastq file  \n",
    "Input file: merged fastq files(.fastq)  \n",
    "Output file: trimmed files(.fq)  \n",
    "\n",
    "As with merge, I also navigated to the analysis/merge/ folder in Terminal and used the following line of code. The output files went into analysis/merge/ but I moved them into analysis/trimgalore/. New output file name convention is TORN_Pool_#_S#_R#.fastq."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "(for i in *_R1.fastq; do /Applications/bio-tools/TrimGalore-0.5.0/trim_galore --path_to_cutadapt /Applications/bio-tools/cutadapt-master/cutadapt -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC -a2 AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT --paired ${i%_R1.fastq}_R1.fastq ${i%_R1.fastq}_R2.fastq; done) >& trim.log.txt"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
